// microtemplates/index.js
/** (github.com/dynamic-deadfunction-detector)
* Instrumentation log function used by the instrumenter
* note that the data object is already stringified.
*/
var logHistory = [];
function instrumentation_log(data) {
    
                function exists(e) { return e.file == jData.file && e.range[0] == jData.range[0] && e.range[1] == jData.range[1]; }
                var jData = JSON.parse(data);
                if (logHistory.some(exists)){ return; }
                logHistory.push(jData);
    fetch("http://127.0.0.1:8004/alivefunction", {
                method: "POST",
                headers: { "Accept": "application/json", "Content-Type": "application/json" },
                body: data
            }).then((response) => { });
}


// Simple JavaScript Templating
// Paul Miller (http://paulmillr.com)
// http://underscorejs.org
// (c) 2009-2013 Jeremy Ashkenas, DocumentCloud and Investigative Reporters & Editors
(function(globals) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[202,3798],"range":[184,3798],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":7,"label":"exoskeleton"}');
  // By default, Underscore uses ERB-style template delimiters, change the
  // following template settings to use alternative delimiters.
  var settings = {
    evaluate    : /<%([\s\S]+?)%>/g,
    interpolate : /<%=([\s\S]+?)%>/g,
    escape      : /<%-([\s\S]+?)%>/g
  };

  // When customizing `templateSettings`, if you don't want to define an
  // interpolation, evaluation or escaping regex, we need one that is
  // guaranteed not to match.
  var noMatch = /(.)^/;

  // Certain characters need to be escaped so that they can be put into a
  // string literal.
  var escapes = {
    "'":      "'",
    '\\':     '\\',
    '\r':     'r',
    '\n':     'n',
    '\t':     't',
    '\u2028': 'u2028',
    '\u2029': 'u2029'
  };

  var escaper = /\\|'|\r|\n|\t|\u2028|\u2029/g;

  // List of HTML entities for escaping.
  var htmlEntities = {
    '&': '&amp;',
    '<': '&lt;',
    '>': '&gt;',
    '"': '&quot;',
    "'": '&#x27;'
  };

  var entityRe = new RegExp('[&<>"\']', 'g');

  var escapeExpr = function(string) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1229,1372],"range":[1212,1372],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":1,"label":"exoskeleton"}');
    if (string == null) return '';
    return ('' + string).replace(entityRe, function(match) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1325,1366],"range":[1309,1366],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":0,"label":"exoskeleton"}');
      return htmlEntities[match];
    });
  };

  var counter = 0;

  // JavaScript micro-templating, similar to John Resig's implementation.
  // Underscore templating handles arbitrary delimiters, preserves whitespace,
  // and correctly escapes quotes within interpolated code.
  var tmpl = function(text, data) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1642,3480],"range":[1621,3480],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":5,"label":"exoskeleton"}');
    var render;

    // Combine delimiters into one regular expression via alternation.
    var matcher = new RegExp([
      (settings.escape || noMatch).source,
      (settings.interpolate || noMatch).source,
      (settings.evaluate || noMatch).source
    ].join('|') + '|$', 'g');

    // Compile the template source, escaping string literals appropriately.
    var index = 0;
    var source = "__p+='";
    text.replace(matcher, function(match, escape, interpolate, evaluate, offset) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[2132,2609],"range":[2077,2609],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":3,"label":"exoskeleton"}');
      source += text.slice(index, offset)
        .replace(escaper, function(match) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[2218,2251],"range":[2202,2251],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":2,"label":"exoskeleton"}'); return '\\' + escapes[match]; });

      if (escape) {
        source += "'+\n((__t=(" + escape + "))==null?'':escapeExpr(__t))+\n'";
      }
      if (interpolate) {
        source += "'+\n((__t=(" + interpolate + "))==null?'':__t)+\n'";
      }
      if (evaluate) {
        source += "';\n" + evaluate + "\n__p+='";
      }
      index = offset + match.length;
      return match;
    });
    source += "';\n";

    // If a variable is not specified, place data values in local scope.
    if (!settings.variable) source = 'with(obj||{}){\n' + source + '}\n';

    source = "var __t,__p='',__j=Array.prototype.join," +
      "print=function(){__p+=__j.call(arguments,'');};\n" +
      source + "return __p;\n//# sourceURL=/microtemplates/source[" + counter++ + "]";

    try {
      render = new Function(settings.variable || 'obj', 'escapeExpr', source);
    } catch (e) {
      e.source = source;
      throw e;
    }

    if (data) return render(data, escapeExpr);
    var template = function(data) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[3224,3281],"range":[3209,3281],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":4,"label":"exoskeleton"}');
      return render.call(this, data, escapeExpr);
    };

    // Provide the compiled function source as a convenience for precompilation.
    template.source = 'function(' + (settings.variable || 'obj') + '){\n' + source + '}';

    return template;
  };
  tmpl.settings = settings;

  if (typeof define !== 'undefined' && define.amd) {
    define([], function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[3591,3617],"range":[3579,3617],"file":"todomvc/examples.lacunized.instrumented/exoskeleton/node_modules/microtemplates/index.js","index":6,"label":"exoskeleton"}');
      return tmpl;
    }); // RequireJS
  } else if (typeof module !== 'undefined' && module.exports) {
    module.exports = tmpl; // CommonJS
  } else {
    globals.microtemplate = tmpl; // <script>
  }
})(this);
