// handlebars/dist/handlebars.js
/** (github.com/dynamic-deadfunction-detector)
* Instrumentation log function used by the instrumenter
* note that the data object is already stringified.
*/
var logHistory = [];
function instrumentation_log(data) {
    
                function exists(e) { return e.file == jData.file && e.range[0] == jData.range[0] && e.range[1] == jData.range[1]; }
                var jData = JSON.parse(data);
                if (logHistory.some(exists)){ return; }
                logHistory.push(jData);
    fetch("http://127.0.0.1:8004/alivefunction", {
                method: "POST",
                headers: { "Accept": "application/json", "Content-Type": "application/json" },
                body: data
            }).then((response) => { });
}


/*!

 handlebars v1.3.0

Copyright (C) 2011 by Yehuda Katz

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.

@license
*/
/* exported Handlebars */
var Handlebars = (function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1151,89662],"range":[1140,89662],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":191,"label":"dijon"}');
// handlebars/safe-string.js
var __module4__ = (function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1212,1486],"range":[1201,1486],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":2,"label":"dijon"}');
  "use strict";
  var __exports__;
  // Build out our basic SafeString type
  function SafeString(string) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[1320,1351],"range":[1292,1351],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":0,"label":"dijon"}');
    this.string = string;
  }

  SafeString.prototype.toString = function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1398,1432],"range":[1387,1432],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":1,"label":"dijon"}');
    return "" + this.string;
  };

  __exports__ = SafeString;
  return __exports__;
})();

// handlebars/utils.js
var __module3__ = (function(__dependency1__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[1560,3728],"range":[1534,3728],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":10,"label":"dijon"}');
  "use strict";
  var __exports__ = {};
  /*jshint -W004 */
  var SafeString = __dependency1__;

  var escape = {
    "&": "&amp;",
    "<": "&lt;",
    ">": "&gt;",
    '"': "&quot;",
    "'": "&#x27;",
    "`": "&#x60;"
  };

  var badChars = /[&<>"'`]/g;
  var possible = /[&<>"'`]/;

  function escapeChar(chr) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[1877,1917],"range":[1852,1917],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":3,"label":"dijon"}');
    return escape[chr] || "&amp;";
  }

  function extend(obj, value) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[1949,2088],"range":[1921,2088],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":4,"label":"dijon"}');
    for(var key in value) {
      if(Object.prototype.hasOwnProperty.call(value, key)) {
        obj[key] = value[key];
      }
    }
  }

  __exports__.extend = extend;var toString = Object.prototype.toString;
  __exports__.toString = toString;
  // Sourced from lodash
  // https://github.com/bestiejs/lodash/blob/master/LICENSE.txt
  var isFunction = function(value) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[2321,2366],"range":[2305,2366],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":5,"label":"dijon"}');
    return typeof value === 'function';
  };
  // fallback for older versions of Chrome and Safari
  if (isFunction(/x/)) {
    isFunction = function(value) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[2480,2577],"range":[2464,2577],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":6,"label":"dijon"}');
      return typeof value === 'function' && toString.call(value) === '[object Function]';
    };
  }
  var isFunction;
  __exports__.isFunction = isFunction;
  var isArray = Array.isArray || function(value) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[2689,2795],"range":[2673,2795],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":7,"label":"dijon"}');
    return (value && typeof value === 'object') ? toString.call(value) === '[object Array]' : false;
  };
  __exports__.isArray = isArray;

  function escapeExpression(string) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[2867,3424],"range":[2833,3424],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":8,"label":"dijon"}');
    // don't escape SafeStrings, since they're already safe
    if (string instanceof SafeString) {
      return string.toString();
    } else if (!string && string !== 0) {
      return "";
    }

    // Force a string conversion as this will be done by the append regardless and
    // the regex test will do this transparently behind the scenes, causing issues if
    // an object's to string has escaped characters in it.
    string = "" + string;

    if(!possible.test(string)) { return string; }
    return string.replace(badChars, escapeChar);
  }

  __exports__.escapeExpression = escapeExpression;function isEmpty(value) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[3500,3670],"range":[3476,3670],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":9,"label":"dijon"}');
    if (!value && value !== 0) {
      return true;
    } else if (isArray(value) && value.length === 0) {
      return true;
    } else {
      return false;
    }
  }

  __exports__.isEmpty = isEmpty;
  return __exports__;
})(__module4__);

// handlebars/exception.js
var __module5__ = (function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[3802,4586],"range":[3791,4586],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":12,"label":"dijon"}');
  "use strict";
  var __exports__;

  var errorProps = ['description', 'fileName', 'lineNumber', 'message', 'name', 'number', 'stack'];

  function Exception(message, node) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[3977,4496],"range":[3943,4496],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":11,"label":"dijon"}');
    var line;
    if (node && node.firstLine) {
      line = node.firstLine;

      message += ' - ' + line + ':' + node.firstColumn;
    }

    var tmp = Error.prototype.constructor.call(this, message);

    // Unfortunately errors are not enumerable in Chrome (at least), so `for prop in tmp` doesn't work.
    for (var idx = 0; idx < errorProps.length; idx++) {
      this[errorProps[idx]] = tmp[errorProps[idx]];
    }

    if (line) {
      this.lineNumber = line;
      this.column = node.firstColumn;
    }
  }

  Exception.prototype = new Error();

  __exports__ = Exception;
  return __exports__;
})();

// handlebars/base.js
var __module2__ = (function(__dependency1__, __dependency2__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[4676,10159],"range":[4633,10159],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":28,"label":"dijon"}');
  "use strict";
  var __exports__ = {};
  var Utils = __dependency1__;
  var Exception = __dependency2__;

  var VERSION = "1.3.0";
  __exports__.VERSION = VERSION;var COMPILER_REVISION = 4;
  __exports__.COMPILER_REVISION = COMPILER_REVISION;
  var REVISION_CHANGES = {
    1: '<= 1.0.rc.2', // 1.0.rc.2 is actually rev2 but doesn't report it
    2: '== 1.0.0-rc.3',
    3: '== 1.0.0-rc.4',
    4: '>= 1.0.0'
  };
  __exports__.REVISION_CHANGES = REVISION_CHANGES;
  var isArray = Utils.isArray,
      isFunction = Utils.isFunction,
      toString = Utils.toString,
      objectType = '[object Object]';

  function HandlebarsEnvironment(helpers, partials) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[5336,5446],"range":[5286,5446],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":13,"label":"dijon"}');
    this.helpers = helpers || {};
    this.partials = partials || {};

    registerDefaultHelpers(this);
  }

  __exports__.HandlebarsEnvironment = HandlebarsEnvironment;HandlebarsEnvironment.prototype = {
    constructor: HandlebarsEnvironment,

    logger: logger,
    log: log,

    registerHelper: function(name, fn, inverse) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[5668,5959],"range":[5640,5959],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":14,"label":"dijon"}');
      if (toString.call(name) === objectType) {
        if (inverse || fn) { throw new Exception('Arg not supported with multiple helpers'); }
        Utils.extend(this.helpers, name);
      } else {
        if (inverse) { fn.not = inverse; }
        this.helpers[name] = fn;
      }
    },

    registerPartial: function(name, str) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[6003,6160],"range":[5983,6160],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":15,"label":"dijon"}');
      if (toString.call(name) === objectType) {
        Utils.extend(this.partials,  name);
      } else {
        this.partials[name] = str;
      }
    }
  };

  function registerDefaultHelpers(instance) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[6211,9403],"range":[6169,9403],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":24,"label":"dijon"}');
    instance.registerHelper('helperMissing', function(arg) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[6272,6425],"range":[6258,6425],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":16,"label":"dijon"}');
      if(arguments.length === 2) {
        return undefined;
      } else {
        throw new Exception("Missing helper: '" + arg + "'");
      }
    });

    instance.registerHelper('blockHelperMissing', function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[6506,7029],"range":[6479,7029],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":18,"label":"dijon"}');
      var inverse = options.inverse || function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[6558,6560],"range":[6547,6560],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":17,"label":"dijon"}');}, fn = options.fn;

      if (isFunction(context)) { context = context.call(this); }

      if(context === true) {
        return fn(this);
      } else if(context === false || context == null) {
        return inverse(this);
      } else if (isArray(context)) {
        if(context.length > 0) {
          return instance.helpers.each(context, options);
        } else {
          return inverse(this);
        }
      } else {
        return fn(context);
      }
    });

    instance.registerHelper('each', function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[7096,8139],"range":[7069,8139],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":19,"label":"dijon"}');
      var fn = options.fn, inverse = options.inverse;
      var i = 0, ret = "", data;

      if (isFunction(context)) { context = context.call(this); }

      if (options.data) {
        data = createFrame(options.data);
      }

      if(context && typeof context === 'object') {
        if (isArray(context)) {
          for(var j = context.length; i<j; i++) {
            if (data) {
              data.index = i;
              data.first = (i === 0);
              data.last  = (i === (context.length-1));
            }
            ret = ret + fn(context[i], { data: data });
          }
        } else {
          for(var key in context) {
            if(context.hasOwnProperty(key)) {
              if(data) { 
                data.key = key; 
                data.index = i;
                data.first = (i === 0);
              }
              ret = ret + fn(context[key], {data: data});
              i++;
            }
          }
        }
      }

      if(i === 0){
        ret = inverse(this);
      }

      return ret;
    });

    instance.registerHelper('if', function(conditional, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[8208,8779],"range":[8177,8779],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":20,"label":"dijon"}');
      if (isFunction(conditional)) { conditional = conditional.call(this); }

      // Default behavior is to render the positive path if the value is truthy and not empty.
      // The `includeZero` option may be set to treat the condtional as purely not empty based on the
      // behavior of isEmpty. Effectively this determines if 0 is handled by the positive path or negative.
      if ((!options.hash.includeZero && !conditional) || Utils.isEmpty(conditional)) {
        return options.inverse(this);
      } else {
        return options.fn(this);
      }
    });

    instance.registerHelper('unless', function(conditional, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[8852,8984],"range":[8821,8984],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":21,"label":"dijon"}');
      return instance.helpers['if'].call(this, conditional, {fn: options.inverse, inverse: options.fn, hash: options.hash});
    });

    instance.registerHelper('with', function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[9051,9187],"range":[9024,9187],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":22,"label":"dijon"}');
      if (isFunction(context)) { context = context.call(this); }

      if (!Utils.isEmpty(context)) return options.fn(context);
    });

    instance.registerHelper('log', function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[9253,9397],"range":[9226,9397],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":23,"label":"dijon"}');
      var level = options.data && options.data.level != null ? parseInt(options.data.level, 10) : 1;
      instance.log(level, context);
    });
  }

  var logger = {
    methodMap: { 0: 'debug', 1: 'info', 2: 'warn', 3: 'error' },

    // State enum
    DEBUG: 0,
    INFO: 1,
    WARN: 2,
    ERROR: 3,
    level: 3,

    // can be overridden in the host environment
    log: function(level, obj) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[9654,9871],"range":[9633,9871],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":25,"label":"dijon"}');
      if (logger.level <= level) {
        var method = logger.methodMap[level];
        if (typeof console !== 'undefined' && console[method]) {
          console[method].call(console, obj);
        }
      }
    }
  };
  __exports__.logger = logger;
  function log(level, obj) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[9935,9962],"range":[9910,9962],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":26,"label":"dijon"}'); logger.log(level, obj); }

  __exports__.log = log;var createFrame = function(object) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[10023,10093],"range":[10006,10093],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":27,"label":"dijon"}');
    var obj = {};
    Utils.extend(obj, object);
    return obj;
  };
  __exports__.createFrame = createFrame;
  return __exports__;
})(__module3__, __module5__);

// handlebars/runtime.js
var __module6__ = (function(__dependency1__, __dependency2__, __dependency3__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[10293,15063],"range":[10233,15063],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":41,"label":"dijon"}');
  "use strict";
  var __exports__ = {};
  var Utils = __dependency1__;
  var Exception = __dependency2__;
  var COMPILER_REVISION = __dependency3__.COMPILER_REVISION;
  var REVISION_CHANGES = __dependency3__.REVISION_CHANGES;

  function checkRevision(compilerInfo) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[10561,11514],"range":[10524,11514],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":29,"label":"dijon"}');
    var compilerRevision = compilerInfo && compilerInfo[0] || 1,
        currentRevision = COMPILER_REVISION;

    if (compilerRevision !== currentRevision) {
      if (compilerRevision < currentRevision) {
        var runtimeVersions = REVISION_CHANGES[currentRevision],
            compilerVersions = REVISION_CHANGES[compilerRevision];
        throw new Exception("Template was precompiled with an older version of Handlebars than the current runtime. "+
              "Please update your precompiler to a newer version ("+runtimeVersions+") or downgrade your runtime to an older version ("+compilerVersions+").");
      } else {
        // Use the embedded version info since the runtime doesn't know about this revision yet
        throw new Exception("Template was precompiled with a newer version of Handlebars than the current runtime. "+
              "Please update your runtime to a newer version ("+compilerInfo[1]+").");
      }
    }
  }

  __exports__.checkRevision = checkRevision;// TODO: Remove this line and break up compilePartial

  function template(templateSpec, env) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[11654,13883],"range":[11617,13883],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":34,"label":"dijon"}');
    if (!env) {
      throw new Exception("No environment passed to template");
    }

    // Note: Using env.VM references rather than local var references throughout this section to allow
    // for external users to override these as psuedo-supported APIs.
    var invokePartialWrapper = function(partial, name, context, helpers, partials, data) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[12005,12490],"range":[11947,12490],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":30,"label":"dijon"}');
      var result = env.VM.invokePartial.apply(this, arguments);
      if (result != null) { return result; }

      if (env.compile) {
        var options = { helpers: helpers, partials: partials, data: data };
        partials[name] = env.compile(partial, { data: data !== undefined }, env);
        return partials[name](context, options);
      } else {
        throw new Exception("The partial " + name + " could not be compiled when running in runtime-only mode");
      }
    };

    // Just add water
    var container = {
      escapeExpression: Utils.escapeExpression,
      invokePartial: invokePartialWrapper,
      programs: [],
      program: function(i, fn, data) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[12685,12950],"range":[12663,12950],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":31,"label":"dijon"}');
        var programWrapper = this.programs[i];
        if(data) {
          programWrapper = program(i, fn, data);
        } else if (!programWrapper) {
          programWrapper = this.programs[i] = program(i, fn);
        }
        return programWrapper;
      },
      merge: function(param, common) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[12989,13210],"range":[12965,13210],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":32,"label":"dijon"}');
        var ret = param || common;

        if (param && common && (param !== common)) {
          ret = {};
          Utils.extend(ret, common);
          Utils.extend(ret, param);
        }
        return ret;
      },
      programWithDepth: env.VM.programWithDepth,
      noop: env.VM.noop,
      compilerInfo: null
    };

    return function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[13357,13878],"range":[13330,13878],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":33,"label":"dijon"}');
      options = options || {};
      var namespace = options.partial ? options : env,
          helpers,
          partials;

      if (!options.partial) {
        helpers = options.helpers;
        partials = options.partials;
      }
      var result = templateSpec.call(
            container,
            namespace, context,
            helpers,
            partials,
            options.data);

      if (!options.partial) {
        env.VM.checkRevision(container.compilerInfo);
      }

      return result;
    };
  }

  __exports__.template = template;function programWithDepth(i, fn, data /*, $depth */) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[13972,14262],"range":[13919,14262],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":36,"label":"dijon"}');
    var args = Array.prototype.slice.call(arguments, 3);

    var prog = function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[14074,14188],"range":[14047,14188],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":35,"label":"dijon"}');
      options = options || {};

      return fn.apply(this, [context, options.data || data].concat(args));
    };
    prog.program = i;
    prog.depth = args.length;
    return prog;
  }

  __exports__.programWithDepth = programWithDepth;function program(i, fn, data) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[14344,14539],"range":[14314,14539],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":38,"label":"dijon"}');
    var prog = function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[14388,14475],"range":[14361,14475],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":37,"label":"dijon"}');
      options = options || {};

      return fn(context, options.data || data);
    };
    prog.program = i;
    prog.depth = 0;
    return prog;
  }

  __exports__.program = program;function invokePartial(partial, name, context, helpers, partials, data) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[14645,14935],"range":[14573,14935],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":39,"label":"dijon"}');
    var options = { partial: true, helpers: helpers, partials: partials, data: data };

    if(partial === undefined) {
      throw new Exception("The partial " + name + " could not be found");
    } else if(partial instanceof Function) {
      return partial(context, options);
    }
  }

  __exports__.invokePartial = invokePartial;function noop() {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[14997,15011],"range":[14981,15011],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":40,"label":"dijon"}'); return ""; }

  __exports__.noop = noop;
  return __exports__;
})(__module3__, __module5__, __module2__);

// handlebars.runtime.js
var __module1__ = (function(__dependency1__, __dependency2__, __dependency3__, __dependency4__, __dependency5__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[15244,16152],"range":[15150,16152],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":44,"label":"dijon"}');
  "use strict";
  var __exports__;
  /*globals Handlebars: true */
  var base = __dependency1__;

  // Each of these augment the Handlebars object. No need to setup here.
  // (This is done to easily share code between commonjs and browse envs)
  var SafeString = __dependency2__;
  var Exception = __dependency3__;
  var Utils = __dependency4__;
  var runtime = __dependency5__;

  // For compatibility and usage outside of module systems, make the Handlebars object a namespace
  var create = function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[15752,16038],"range":[15741,16038],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":43,"label":"dijon"}');
    var hb = new base.HandlebarsEnvironment();

    Utils.extend(hb, base);
    hb.SafeString = SafeString;
    hb.Exception = Exception;
    hb.Utils = Utils;

    hb.VM = runtime;
    hb.template = function(spec) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[15969,16017],"range":[15954,16017],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":42,"label":"dijon"}');
      return runtime.template(spec, hb);
    };

    return hb;
  };

  var Handlebars = create();
  Handlebars.create = create;

  __exports__ = Handlebars;
  return __exports__;
})(__module2__, __module4__, __module5__, __module3__, __module6__);

// handlebars/compiler/ast.js
var __module7__ = (function(__dependency1__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[16296,23061],"range":[16270,23061],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":60,"label":"dijon"}');
  "use strict";
  var __exports__;
  var Exception = __dependency1__;

  function LocationInfo(locInfo){instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[16401,16609],"range":[16371,16609],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":45,"label":"dijon"}');
    locInfo = locInfo || {};
    this.firstLine   = locInfo.first_line;
    this.firstColumn = locInfo.first_column;
    this.lastColumn  = locInfo.last_column;
    this.lastLine    = locInfo.last_line;
  }

  var AST = {
    ProgramNode: function(statements, inverseStrip, inverse, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[16695,17748],"range":[16642,17748],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":46,"label":"dijon"}');
      var inverseLocationInfo, firstInverseNode;
      if (arguments.length === 3) {
        locInfo = inverse;
        inverse = null;
      } else if (arguments.length === 2) {
        locInfo = inverseStrip;
        inverseStrip = null;
      }

      LocationInfo.call(this, locInfo);
      this.type = "program";
      this.statements = statements;
      this.strip = {};

      if(inverse) {
        firstInverseNode = inverse[0];
        if (firstInverseNode) {
          inverseLocationInfo = {
            first_line: firstInverseNode.firstLine,
            last_line: firstInverseNode.lastLine,
            last_column: firstInverseNode.lastColumn,
            first_column: firstInverseNode.firstColumn
          };
          this.inverse = new AST.ProgramNode(inverse, inverseStrip, inverseLocationInfo);
        } else {
          this.inverse = new AST.ProgramNode(inverse, inverseStrip);
        }
        this.strip.right = inverseStrip.left;
      } else if (inverseStrip) {
        this.strip.left = inverseStrip.right;
      }
    },

    MustacheNode: function(rawParams, hash, open, strip, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[17817,18762],"range":[17769,18762],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":47,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "mustache";
      this.strip = strip;

      // Open may be a string parsed from the parser or a passed boolean flag
      if (open != null && open.charAt) {
        // Must use charAt to support IE pre-10
        var escapeFlag = open.charAt(3) || open.charAt(2);
        this.escaped = escapeFlag !== '{' && escapeFlag !== '&';
      } else {
        this.escaped = !!open;
      }

      if (rawParams instanceof AST.SexprNode) {
        this.sexpr = rawParams;
      } else {
        // Support old AST API
        this.sexpr = new AST.SexprNode(rawParams, hash);
      }

      this.sexpr.isRoot = true;

      // Support old AST API that stored this info in MustacheNode
      this.id = this.sexpr.id;
      this.params = this.sexpr.params;
      this.hash = this.sexpr.hash;
      this.eligibleHelper = this.sexpr.eligibleHelper;
      this.isHelper = this.sexpr.isHelper;
    },

    SexprNode: function(rawParams, hash, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[18815,19551],"range":[18780,19551],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":48,"label":"dijon"}');
      LocationInfo.call(this, locInfo);

      this.type = "sexpr";
      this.hash = hash;

      var id = this.id = rawParams[0];
      var params = this.params = rawParams.slice(1);

      // a mustache is an eligible helper if:
      // * its id is simple (a single part, not `this` or `..`)
      var eligibleHelper = this.eligibleHelper = id.isSimple;

      // a mustache is definitely a helper if:
      // * it is an eligible helper, and
      // * it has at least one parameter or hash segment
      this.isHelper = eligibleHelper && (params.length || hash);

      // if a mustache is an eligible helper but not a definite
      // helper, it is ambiguous, and will be resolved in a later
      // pass or at runtime.
    },

    PartialNode: function(partialName, context, strip, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[19618,19802],"range":[19571,19802],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":49,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type         = "partial";
      this.partialName  = partialName;
      this.context      = context;
      this.strip = strip;
    },

    BlockNode: function(mustache, program, inverse, close, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[19873,20513],"range":[19820,20513],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":50,"label":"dijon"}');
      LocationInfo.call(this, locInfo);

      if(mustache.sexpr.id.original !== close.path.original) {
        throw new Exception(mustache.sexpr.id.original + " doesn't match " + close.path.original, this);
      }

      this.type = 'block';
      this.mustache = mustache;
      this.program  = program;
      this.inverse  = inverse;

      this.strip = {
        left: mustache.strip.left,
        right: close.strip.right
      };

      (program || inverse).strip.left = mustache.strip.right;
      (inverse || program).strip.right = close.strip.left;

      if (inverse && !program) {
        this.isInverse = true;
      }
    },

    ContentNode: function(string, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[20559,20663],"range":[20533,20663],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":51,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "content";
      this.string = string;
    },

    HashNode: function(pairs, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[20705,20804],"range":[20680,20804],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":52,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "hash";
      this.pairs = pairs;
    },

    IdNode: function(parts, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[20844,21809],"range":[20819,21809],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":53,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "ID";

      var original = "",
          dig = [],
          depth = 0;

      for(var i=0,l=parts.length; i<l; i++) {
        var part = parts[i].part;
        original += (parts[i].separator || '') + part;

        if (part === ".." || part === "." || part === "this") {
          if (dig.length > 0) {
            throw new Exception("Invalid path: " + original, this);
          } else if (part === "..") {
            depth++;
          } else {
            this.isScoped = true;
          }
        } else {
          dig.push(part);
        }
      }

      this.original = original;
      this.parts    = dig;
      this.string   = dig.join('.');
      this.depth    = depth;

      // an ID is simple if it only has one part, and that part is not
      // `..` or `this`.
      this.isSimple = parts.length === 1 && !this.isScoped && depth === 0;

      this.stringModeValue = this.string;
    },

    PartialNameNode: function(name, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[21857,21971],"range":[21833,21971],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":54,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "PARTIAL_NAME";
      this.name = name.original;
    },

    DataNode: function(id, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[22010,22103],"range":[21988,22103],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":55,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "DATA";
      this.id = id;
    },

    StringNode: function(string, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[22148,22306],"range":[22122,22306],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":56,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "STRING";
      this.original =
        this.string =
        this.stringModeValue = string;
    },

    IntegerNode: function(integer, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[22353,22529],"range":[22326,22529],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":57,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "INTEGER";
      this.original =
        this.integer = integer;
      this.stringModeValue = Number(integer);
    },

    BooleanNode: function(bool, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[22573,22719],"range":[22549,22719],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":58,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "BOOLEAN";
      this.bool = bool;
      this.stringModeValue = bool === "true";
    },

    CommentNode: function(comment, locInfo) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[22766,22872],"range":[22739,22872],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":59,"label":"dijon"}');
      LocationInfo.call(this, locInfo);
      this.type = "comment";
      this.comment = comment;
    }
  };

  // Must be exported as an object rather than the root of the module as the jison lexer
  // most modify the object to operate properly.
  __exports__ = AST;
  return __exports__;
})(__module5__);

// handlebars/compiler/parser.js
var __module9__ = (function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[23141,47180],"range":[23130,47180],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":89,"label":"dijon"}');
  "use strict";
  var __exports__;
  /* jshint ignore:start */
  /* Jison generated parser */
  var handlebars = (function(){instrumentation_log('{"type":"FunctionExpression","bodyRange":[23267,47101],"range":[23257,47101],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":88,"label":"dijon"}');
  var parser = {trace: function trace() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[23309,23312],"range":[23292,23312],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":61,"label":"dijon"}'); },
  yy: {},
  symbols_: {"error":2,"root":3,"statements":4,"EOF":5,"program":6,"simpleInverse":7,"statement":8,"openInverse":9,"closeBlock":10,"openBlock":11,"mustache":12,"partial":13,"CONTENT":14,"COMMENT":15,"OPEN_BLOCK":16,"sexpr":17,"CLOSE":18,"OPEN_INVERSE":19,"OPEN_ENDBLOCK":20,"path":21,"OPEN":22,"OPEN_UNESCAPED":23,"CLOSE_UNESCAPED":24,"OPEN_PARTIAL":25,"partialName":26,"partial_option0":27,"sexpr_repetition0":28,"sexpr_option0":29,"dataName":30,"param":31,"STRING":32,"INTEGER":33,"BOOLEAN":34,"OPEN_SEXPR":35,"CLOSE_SEXPR":36,"hash":37,"hash_repetition_plus0":38,"hashSegment":39,"ID":40,"EQUALS":41,"DATA":42,"pathSegments":43,"SEP":44,"$accept":0,"$end":1},
  terminals_: {2:"error",5:"EOF",14:"CONTENT",15:"COMMENT",16:"OPEN_BLOCK",18:"CLOSE",19:"OPEN_INVERSE",20:"OPEN_ENDBLOCK",22:"OPEN",23:"OPEN_UNESCAPED",24:"CLOSE_UNESCAPED",25:"OPEN_PARTIAL",32:"STRING",33:"INTEGER",34:"BOOLEAN",35:"OPEN_SEXPR",36:"CLOSE_SEXPR",40:"ID",41:"EQUALS",42:"DATA",44:"SEP"},
  productions_: [0,[3,2],[3,1],[6,2],[6,3],[6,2],[6,1],[6,1],[6,0],[4,1],[4,2],[8,3],[8,3],[8,1],[8,1],[8,1],[8,1],[11,3],[9,3],[10,3],[12,3],[12,3],[13,4],[7,2],[17,3],[17,1],[31,1],[31,1],[31,1],[31,1],[31,1],[31,3],[37,1],[39,3],[26,1],[26,1],[26,1],[30,2],[21,1],[43,3],[43,1],[27,0],[27,1],[28,0],[28,2],[29,0],[29,1],[38,1],[38,2]],
  performAction: function anonymous(yytext,yyleng,yylineno,yy,yystate,$$,_$) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[24707,27794],"range":[24647,27794],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":62,"label":"dijon"}');

  var $0 = $$.length - 1;
  switch (yystate) {
  case 1: return new yy.ProgramNode($$[$0-1], this._$); 
  break;
  case 2: return new yy.ProgramNode([], this._$); 
  break;
  case 3:this.$ = new yy.ProgramNode([], $$[$0-1], $$[$0], this._$);
  break;
  case 4:this.$ = new yy.ProgramNode($$[$0-2], $$[$0-1], $$[$0], this._$);
  break;
  case 5:this.$ = new yy.ProgramNode($$[$0-1], $$[$0], [], this._$);
  break;
  case 6:this.$ = new yy.ProgramNode($$[$0], this._$);
  break;
  case 7:this.$ = new yy.ProgramNode([], this._$);
  break;
  case 8:this.$ = new yy.ProgramNode([], this._$);
  break;
  case 9:this.$ = [$$[$0]];
  break;
  case 10: $$[$0-1].push($$[$0]); this.$ = $$[$0-1]; 
  break;
  case 11:this.$ = new yy.BlockNode($$[$0-2], $$[$0-1].inverse, $$[$0-1], $$[$0], this._$);
  break;
  case 12:this.$ = new yy.BlockNode($$[$0-2], $$[$0-1], $$[$0-1].inverse, $$[$0], this._$);
  break;
  case 13:this.$ = $$[$0];
  break;
  case 14:this.$ = $$[$0];
  break;
  case 15:this.$ = new yy.ContentNode($$[$0], this._$);
  break;
  case 16:this.$ = new yy.CommentNode($$[$0], this._$);
  break;
  case 17:this.$ = new yy.MustacheNode($$[$0-1], null, $$[$0-2], stripFlags($$[$0-2], $$[$0]), this._$);
  break;
  case 18:this.$ = new yy.MustacheNode($$[$0-1], null, $$[$0-2], stripFlags($$[$0-2], $$[$0]), this._$);
  break;
  case 19:this.$ = {path: $$[$0-1], strip: stripFlags($$[$0-2], $$[$0])};
  break;
  case 20:this.$ = new yy.MustacheNode($$[$0-1], null, $$[$0-2], stripFlags($$[$0-2], $$[$0]), this._$);
  break;
  case 21:this.$ = new yy.MustacheNode($$[$0-1], null, $$[$0-2], stripFlags($$[$0-2], $$[$0]), this._$);
  break;
  case 22:this.$ = new yy.PartialNode($$[$0-2], $$[$0-1], stripFlags($$[$0-3], $$[$0]), this._$);
  break;
  case 23:this.$ = stripFlags($$[$0-1], $$[$0]);
  break;
  case 24:this.$ = new yy.SexprNode([$$[$0-2]].concat($$[$0-1]), $$[$0], this._$);
  break;
  case 25:this.$ = new yy.SexprNode([$$[$0]], null, this._$);
  break;
  case 26:this.$ = $$[$0];
  break;
  case 27:this.$ = new yy.StringNode($$[$0], this._$);
  break;
  case 28:this.$ = new yy.IntegerNode($$[$0], this._$);
  break;
  case 29:this.$ = new yy.BooleanNode($$[$0], this._$);
  break;
  case 30:this.$ = $$[$0];
  break;
  case 31:$$[$0-1].isHelper = true; this.$ = $$[$0-1];
  break;
  case 32:this.$ = new yy.HashNode($$[$0], this._$);
  break;
  case 33:this.$ = [$$[$0-2], $$[$0]];
  break;
  case 34:this.$ = new yy.PartialNameNode($$[$0], this._$);
  break;
  case 35:this.$ = new yy.PartialNameNode(new yy.StringNode($$[$0], this._$), this._$);
  break;
  case 36:this.$ = new yy.PartialNameNode(new yy.IntegerNode($$[$0], this._$));
  break;
  case 37:this.$ = new yy.DataNode($$[$0], this._$);
  break;
  case 38:this.$ = new yy.IdNode($$[$0], this._$);
  break;
  case 39: $$[$0-2].push({part: $$[$0], separator: $$[$0-1]}); this.$ = $$[$0-2]; 
  break;
  case 40:this.$ = [{part: $$[$0]}];
  break;
  case 43:this.$ = [];
  break;
  case 44:$$[$0-1].push($$[$0]);
  break;
  case 47:this.$ = [$$[$0]];
  break;
  case 48:$$[$0-1].push($$[$0]);
  break;
  }
  },
  table: [{3:1,4:2,5:[1,3],8:4,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,11],22:[1,13],23:[1,14],25:[1,15]},{1:[3]},{5:[1,16],8:17,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,11],22:[1,13],23:[1,14],25:[1,15]},{1:[2,2]},{5:[2,9],14:[2,9],15:[2,9],16:[2,9],19:[2,9],20:[2,9],22:[2,9],23:[2,9],25:[2,9]},{4:20,6:18,7:19,8:4,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,21],20:[2,8],22:[1,13],23:[1,14],25:[1,15]},{4:20,6:22,7:19,8:4,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,21],20:[2,8],22:[1,13],23:[1,14],25:[1,15]},{5:[2,13],14:[2,13],15:[2,13],16:[2,13],19:[2,13],20:[2,13],22:[2,13],23:[2,13],25:[2,13]},{5:[2,14],14:[2,14],15:[2,14],16:[2,14],19:[2,14],20:[2,14],22:[2,14],23:[2,14],25:[2,14]},{5:[2,15],14:[2,15],15:[2,15],16:[2,15],19:[2,15],20:[2,15],22:[2,15],23:[2,15],25:[2,15]},{5:[2,16],14:[2,16],15:[2,16],16:[2,16],19:[2,16],20:[2,16],22:[2,16],23:[2,16],25:[2,16]},{17:23,21:24,30:25,40:[1,28],42:[1,27],43:26},{17:29,21:24,30:25,40:[1,28],42:[1,27],43:26},{17:30,21:24,30:25,40:[1,28],42:[1,27],43:26},{17:31,21:24,30:25,40:[1,28],42:[1,27],43:26},{21:33,26:32,32:[1,34],33:[1,35],40:[1,28],43:26},{1:[2,1]},{5:[2,10],14:[2,10],15:[2,10],16:[2,10],19:[2,10],20:[2,10],22:[2,10],23:[2,10],25:[2,10]},{10:36,20:[1,37]},{4:38,8:4,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,11],20:[2,7],22:[1,13],23:[1,14],25:[1,15]},{7:39,8:17,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,21],20:[2,6],22:[1,13],23:[1,14],25:[1,15]},{17:23,18:[1,40],21:24,30:25,40:[1,28],42:[1,27],43:26},{10:41,20:[1,37]},{18:[1,42]},{18:[2,43],24:[2,43],28:43,32:[2,43],33:[2,43],34:[2,43],35:[2,43],36:[2,43],40:[2,43],42:[2,43]},{18:[2,25],24:[2,25],36:[2,25]},{18:[2,38],24:[2,38],32:[2,38],33:[2,38],34:[2,38],35:[2,38],36:[2,38],40:[2,38],42:[2,38],44:[1,44]},{21:45,40:[1,28],43:26},{18:[2,40],24:[2,40],32:[2,40],33:[2,40],34:[2,40],35:[2,40],36:[2,40],40:[2,40],42:[2,40],44:[2,40]},{18:[1,46]},{18:[1,47]},{24:[1,48]},{18:[2,41],21:50,27:49,40:[1,28],43:26},{18:[2,34],40:[2,34]},{18:[2,35],40:[2,35]},{18:[2,36],40:[2,36]},{5:[2,11],14:[2,11],15:[2,11],16:[2,11],19:[2,11],20:[2,11],22:[2,11],23:[2,11],25:[2,11]},{21:51,40:[1,28],43:26},{8:17,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,11],20:[2,3],22:[1,13],23:[1,14],25:[1,15]},{4:52,8:4,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,11],20:[2,5],22:[1,13],23:[1,14],25:[1,15]},{14:[2,23],15:[2,23],16:[2,23],19:[2,23],20:[2,23],22:[2,23],23:[2,23],25:[2,23]},{5:[2,12],14:[2,12],15:[2,12],16:[2,12],19:[2,12],20:[2,12],22:[2,12],23:[2,12],25:[2,12]},{14:[2,18],15:[2,18],16:[2,18],19:[2,18],20:[2,18],22:[2,18],23:[2,18],25:[2,18]},{18:[2,45],21:56,24:[2,45],29:53,30:60,31:54,32:[1,57],33:[1,58],34:[1,59],35:[1,61],36:[2,45],37:55,38:62,39:63,40:[1,64],42:[1,27],43:26},{40:[1,65]},{18:[2,37],24:[2,37],32:[2,37],33:[2,37],34:[2,37],35:[2,37],36:[2,37],40:[2,37],42:[2,37]},{14:[2,17],15:[2,17],16:[2,17],19:[2,17],20:[2,17],22:[2,17],23:[2,17],25:[2,17]},{5:[2,20],14:[2,20],15:[2,20],16:[2,20],19:[2,20],20:[2,20],22:[2,20],23:[2,20],25:[2,20]},{5:[2,21],14:[2,21],15:[2,21],16:[2,21],19:[2,21],20:[2,21],22:[2,21],23:[2,21],25:[2,21]},{18:[1,66]},{18:[2,42]},{18:[1,67]},{8:17,9:5,11:6,12:7,13:8,14:[1,9],15:[1,10],16:[1,12],19:[1,11],20:[2,4],22:[1,13],23:[1,14],25:[1,15]},{18:[2,24],24:[2,24],36:[2,24]},{18:[2,44],24:[2,44],32:[2,44],33:[2,44],34:[2,44],35:[2,44],36:[2,44],40:[2,44],42:[2,44]},{18:[2,46],24:[2,46],36:[2,46]},{18:[2,26],24:[2,26],32:[2,26],33:[2,26],34:[2,26],35:[2,26],36:[2,26],40:[2,26],42:[2,26]},{18:[2,27],24:[2,27],32:[2,27],33:[2,27],34:[2,27],35:[2,27],36:[2,27],40:[2,27],42:[2,27]},{18:[2,28],24:[2,28],32:[2,28],33:[2,28],34:[2,28],35:[2,28],36:[2,28],40:[2,28],42:[2,28]},{18:[2,29],24:[2,29],32:[2,29],33:[2,29],34:[2,29],35:[2,29],36:[2,29],40:[2,29],42:[2,29]},{18:[2,30],24:[2,30],32:[2,30],33:[2,30],34:[2,30],35:[2,30],36:[2,30],40:[2,30],42:[2,30]},{17:68,21:24,30:25,40:[1,28],42:[1,27],43:26},{18:[2,32],24:[2,32],36:[2,32],39:69,40:[1,70]},{18:[2,47],24:[2,47],36:[2,47],40:[2,47]},{18:[2,40],24:[2,40],32:[2,40],33:[2,40],34:[2,40],35:[2,40],36:[2,40],40:[2,40],41:[1,71],42:[2,40],44:[2,40]},{18:[2,39],24:[2,39],32:[2,39],33:[2,39],34:[2,39],35:[2,39],36:[2,39],40:[2,39],42:[2,39],44:[2,39]},{5:[2,22],14:[2,22],15:[2,22],16:[2,22],19:[2,22],20:[2,22],22:[2,22],23:[2,22],25:[2,22]},{5:[2,19],14:[2,19],15:[2,19],16:[2,19],19:[2,19],20:[2,19],22:[2,19],23:[2,19],25:[2,19]},{36:[1,72]},{18:[2,48],24:[2,48],36:[2,48],40:[2,48]},{41:[1,71]},{21:56,30:60,31:73,32:[1,57],33:[1,58],34:[1,59],35:[1,61],40:[1,28],42:[1,27],43:26},{18:[2,31],24:[2,31],32:[2,31],33:[2,31],34:[2,31],35:[2,31],36:[2,31],40:[2,31],42:[2,31]},{18:[2,33],24:[2,33],36:[2,33],40:[2,33]}],
  defaultActions: {3:[2,2],16:[2,1],50:[2,42]},
  parseError: function parseError(str, hash) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[32635,32668],"range":[32604,32668],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":63,"label":"dijon"}');
      throw new Error(str);
  },
  parse: function parse(input) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[32701,37386],"range":[32679,37386],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":66,"label":"dijon"}');
      var self = this, stack = [0], vstack = [null], lstack = [], table = this.table, yytext = "", yylineno = 0, yyleng = 0, recovering = 0, TERROR = 2, EOF = 1;
      this.lexer.setInput(input);
      this.lexer.yy = this.yy;
      this.yy.lexer = this.lexer;
      this.yy.parser = this;
      if (typeof this.lexer.yylloc == "undefined")
          this.lexer.yylloc = {};
      var yyloc = this.lexer.yylloc;
      lstack.push(yyloc);
      var ranges = this.lexer.options && this.lexer.options.ranges;
      if (typeof this.yy.parseError === "function")
          this.parseError = this.yy.parseError;
      function popStack(n) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[33336,33482],"range":[33315,33482],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":64,"label":"dijon"}');
          stack.length = stack.length - 2 * n;
          vstack.length = vstack.length - n;
          lstack.length = lstack.length - n;
      }
      function lex() {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[33504,33707],"range":[33489,33707],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":65,"label":"dijon"}');
          var token;
          token = self.lexer.lex() || 1;
          if (typeof token !== "number") {
              token = self.symbols_[token] || token;
          }
          return token;
      }
      var symbol, preErrorSymbol, state, action, a, r, yyval = {}, p, len, newState, expected;
      while (true) {
          state = stack[stack.length - 1];
          if (this.defaultActions[state]) {
              action = this.defaultActions[state];
          } else {
              if (symbol === null || typeof symbol == "undefined") {
                  symbol = lex();
              }
              action = table[state] && table[state][symbol];
          }
          if (typeof action === "undefined" || !action.length || !action[0]) {
              var errStr = "";
              if (!recovering) {
                  expected = [];
                  for (p in table[state])
                      if (this.terminals_[p] && p > 2) {
                          expected.push("'" + this.terminals_[p] + "'");
                      }
                  if (this.lexer.showPosition) {
                      errStr = "Parse error on line " + (yylineno + 1) + ":\n" + this.lexer.showPosition() + "\nExpecting " + expected.join(", ") + ", got '" + (this.terminals_[symbol] || symbol) + "'";
                  } else {
                      errStr = "Parse error on line " + (yylineno + 1) + ": Unexpected " + (symbol == 1?"end of input":"'" + (this.terminals_[symbol] || symbol) + "'");
                  }
                  this.parseError(errStr, {text: this.lexer.match, token: this.terminals_[symbol] || symbol, line: this.lexer.yylineno, loc: yyloc, expected: expected});
              }
          }
          if (action[0] instanceof Array && action.length > 1) {
              throw new Error("Parse Error: multiple actions possible at state: " + state + ", token: " + symbol);
          }
          switch (action[0]) {
          case 1:
              stack.push(symbol);
              vstack.push(this.lexer.yytext);
              lstack.push(this.lexer.yylloc);
              stack.push(action[1]);
              symbol = null;
              if (!preErrorSymbol) {
                  yyleng = this.lexer.yyleng;
                  yytext = this.lexer.yytext;
                  yylineno = this.lexer.yylineno;
                  yyloc = this.lexer.yylloc;
                  if (recovering > 0)
                      recovering--;
              } else {
                  symbol = preErrorSymbol;
                  preErrorSymbol = null;
              }
              break;
          case 2:
              len = this.productions_[action[1]][1];
              yyval.$ = vstack[vstack.length - len];
              yyval._$ = {first_line: lstack[lstack.length - (len || 1)].first_line, last_line: lstack[lstack.length - 1].last_line, first_column: lstack[lstack.length - (len || 1)].first_column, last_column: lstack[lstack.length - 1].last_column};
              if (ranges) {
                  yyval._$.range = [lstack[lstack.length - (len || 1)].range[0], lstack[lstack.length - 1].range[1]];
              }
              r = this.performAction.call(yyval, yytext, yyleng, yylineno, this.yy, action[1], vstack, lstack);
              if (typeof r !== "undefined") {
                  return r;
              }
              if (len) {
                  stack = stack.slice(0, -1 * len * 2);
                  vstack = vstack.slice(0, -1 * len);
                  lstack = lstack.slice(0, -1 * len);
              }
              stack.push(this.productions_[action[1]][0]);
              vstack.push(yyval.$);
              lstack.push(yyval._$);
              newState = table[stack[stack.length - 2]][stack[stack.length - 1]];
              stack.push(newState);
              break;
          case 3:
              return true;
          }
      }
      return true;
  }
  };


  function stripFlags(open, close) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[37429,37554],"range":[37396,37554],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":67,"label":"dijon"}');
    return {
      left: open.charAt(2) === '~',
      right: close.charAt(0) === '~' || close.charAt(1) === '~'
    };
  }

  /* Jison generated lexer */
  var lexer = (function(){instrumentation_log('{"type":"FunctionExpression","bodyRange":[37611,46961],"range":[37601,46961],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":86,"label":"dijon"}');
  var lexer = ({EOF:1,
  parseError:function parseError(str, hash) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[37680,37840],"range":[37649,37840],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":68,"label":"dijon"}');
          if (this.yy.parser) {
              this.yy.parser.parseError(str, hash);
          } else {
              throw new Error(str);
          }
      },
  setInput:function (input) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[37870,38302],"range":[37853,38302],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":69,"label":"dijon"}');
          this._input = input;
          this._more = this._less = this.done = false;
          this.yylineno = this.yyleng = 0;
          this.yytext = this.matched = this.match = '';
          this.conditionStack = ['INITIAL'];
          this.yylloc = {first_line:1,first_column:0,last_line:1,last_column:0};
          if (this.options.ranges) this.yylloc.range = [0,0];
          this.offset = 0;
          return this;
      },
  input:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[38324,38848],"range":[38312,38848],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":70,"label":"dijon"}');
          var ch = this._input[0];
          this.yytext += ch;
          this.yyleng++;
          this.offset++;
          this.match += ch;
          this.matched += ch;
          var lines = ch.match(/(?:\r\n?|\n).*/g);
          if (lines) {
              this.yylineno++;
              this.yylloc.last_line++;
          } else {
              this.yylloc.last_column++;
          }
          if (this.options.ranges) this.yylloc.range[1]++;

          this._input = this._input.slice(1);
          return ch;
      },
  unput:function (ch) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[38872,39979],"range":[38858,39979],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":71,"label":"dijon"}');
          var len = ch.length;
          var lines = ch.split(/(?:\r\n?|\n)/g);

          this._input = ch + this._input;
          this.yytext = this.yytext.substr(0, this.yytext.length-len-1);
          //this.yyleng -= len;
          this.offset -= len;
          var oldLines = this.match.split(/(?:\r\n?|\n)/g);
          this.match = this.match.substr(0, this.match.length-1);
          this.matched = this.matched.substr(0, this.matched.length-1);

          if (lines.length-1) this.yylineno -= lines.length-1;
          var r = this.yylloc.range;

          this.yylloc = {first_line: this.yylloc.first_line,
            last_line: this.yylineno+1,
            first_column: this.yylloc.first_column,
            last_column: lines ?
                (lines.length === oldLines.length ? this.yylloc.first_column : 0) + oldLines[oldLines.length - lines.length].length - lines[0].length:
                this.yylloc.first_column - len
            };

          if (this.options.ranges) {
              this.yylloc.range = [r[0], r[0] + this.yyleng - len];
          }
          return this;
      },
  more:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[40000,40061],"range":[39988,40061],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":72,"label":"dijon"}');
          this._more = true;
          return this;
      },
  less:function (n) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[40083,40135],"range":[40070,40135],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":73,"label":"dijon"}');
          this.unput(this.match.slice(n));
      },
  pastInput:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[40161,40342],"range":[40149,40342],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":74,"label":"dijon"}');
          var past = this.matched.substr(0, this.matched.length - this.match.length);
          return (past.length > 20 ? '...':'') + past.substr(-20).replace(/\n/g, "");
      },
  upcomingInput:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[40372,40608],"range":[40360,40608],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":75,"label":"dijon"}');
          var next = this.match;
          if (next.length < 20) {
              next += this._input.substr(0, 20-next.length);
          }
          return (next.substr(0,20)+(next.length > 20 ? '...':'')).replace(/\n/g, "");
      },
  showPosition:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[40637,40799],"range":[40625,40799],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":76,"label":"dijon"}');
          var pre = this.pastInput();
          var c = new Array(pre.length + 1).join("-");
          return pre + this.upcomingInput() + "\n" + c+"^";
      },
  next:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[40820,43065],"range":[40808,43065],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":77,"label":"dijon"}');
          if (this.done) {
              return this.EOF;
          }
          if (!this._input) this.done = true;

          var token,
              match,
              tempMatch,
              index,
              col,
              lines;
          if (!this._more) {
              this.yytext = '';
              this.match = '';
          }
          var rules = this._currentRules();
          for (var i=0;i < rules.length; i++) {
              tempMatch = this._input.match(this.rules[rules[i]]);
              if (tempMatch && (!match || tempMatch[0].length > match[0].length)) {
                  match = tempMatch;
                  index = i;
                  if (!this.options.flex) break;
              }
          }
          if (match) {
              lines = match[0].match(/(?:\r\n?|\n).*/g);
              if (lines) this.yylineno += lines.length;
              this.yylloc = {first_line: this.yylloc.last_line,
                             last_line: this.yylineno+1,
                             first_column: this.yylloc.last_column,
                             last_column: lines ? lines[lines.length-1].length-lines[lines.length-1].match(/\r?\n?/)[0].length : this.yylloc.last_column + match[0].length};
              this.yytext += match[0];
              this.match += match[0];
              this.matches = match;
              this.yyleng = this.yytext.length;
              if (this.options.ranges) {
                  this.yylloc.range = [this.offset, this.offset += this.yyleng];
              }
              this._more = false;
              this._input = this._input.slice(match[0].length);
              this.matched += match[0];
              token = this.performAction.call(this, this.yy, this, rules[index],this.conditionStack[this.conditionStack.length-1]);
              if (this.done && this._input) this.done = false;
              if (token) return token;
              else return;
          }
          if (this._input === "") {
              return this.EOF;
          } else {
              return this.parseError('Lexical error on line '+(this.yylineno+1)+'. Unrecognized text.\n'+this.showPosition(),
                      {text: "", token: null, line: this.yylineno});
          }
      },
  lex:function lex() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43088,43258],"range":[43073,43258],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":78,"label":"dijon"}');
          var r = this.next();
          if (typeof r !== 'undefined') {
              return r;
          } else {
              return this.lex();
          }
      },
  begin:function begin(condition) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43294,43350],"range":[43268,43350],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":79,"label":"dijon"}');
          this.conditionStack.push(condition);
      },
  popState:function popState() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43383,43436],"range":[43363,43436],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":80,"label":"dijon"}');
          return this.conditionStack.pop();
      },
  _currentRules:function _currentRules() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43479,43579],"range":[43454,43579],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":81,"label":"dijon"}');
          return this.conditions[this.conditionStack[this.conditionStack.length-1]].rules;
      },
  topState:function () {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43604,43681],"range":[43592,43681],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":82,"label":"dijon"}');
          return this.conditionStack[this.conditionStack.length-2];
      },
  pushState:function begin(condition) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43721,43763],"range":[43695,43763],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":83,"label":"dijon"}');
          this.begin(condition);
      }});
  lexer.options = {};
  lexer.performAction = function anonymous(yy,yy_,$avoiding_name_collisions,YY_START) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[43875,46009],"range":[43813,46009],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":85,"label":"dijon"}');


  function strip(start, end) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[43908,43979],"range":[43881,43979],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":84,"label":"dijon"}');
    return yy_.yytext = yy_.yytext.substr(start, yy_.yyleng-end);
  }


  var YYSTATE=YY_START
  switch($avoiding_name_collisions) {
  case 0:
                                     if(yy_.yytext.slice(-2) === "\\\\") {
                                       strip(0,1);
                                       this.begin("mu");
                                     } else if(yy_.yytext.slice(-1) === "\\") {
                                       strip(0,1);
                                       this.begin("emu");
                                     } else {
                                       this.begin("mu");
                                     }
                                     if(yy_.yytext) return 14;
                                   
  break;
  case 1:return 14;
  break;
  case 2:
                                     this.popState();
                                     return 14;
                                   
  break;
  case 3:strip(0,4); this.popState(); return 15;
  break;
  case 4:return 35;
  break;
  case 5:return 36;
  break;
  case 6:return 25;
  break;
  case 7:return 16;
  break;
  case 8:return 20;
  break;
  case 9:return 19;
  break;
  case 10:return 19;
  break;
  case 11:return 23;
  break;
  case 12:return 22;
  break;
  case 13:this.popState(); this.begin('com');
  break;
  case 14:strip(3,5); this.popState(); return 15;
  break;
  case 15:return 22;
  break;
  case 16:return 41;
  break;
  case 17:return 40;
  break;
  case 18:return 40;
  break;
  case 19:return 44;
  break;
  case 20:// ignore whitespace
  break;
  case 21:this.popState(); return 24;
  break;
  case 22:this.popState(); return 18;
  break;
  case 23:yy_.yytext = strip(1,2).replace(/\\"/g,'"'); return 32;
  break;
  case 24:yy_.yytext = strip(1,2).replace(/\\'/g,"'"); return 32;
  break;
  case 25:return 42;
  break;
  case 26:return 34;
  break;
  case 27:return 34;
  break;
  case 28:return 33;
  break;
  case 29:return 40;
  break;
  case 30:yy_.yytext = strip(1,2); return 40;
  break;
  case 31:return 'INVALID';
  break;
  case 32:return 5;
  break;
  }
  };
  lexer.rules = [/^(?:[^\x00]*?(?=(\{\{)))/,/^(?:[^\x00]+)/,/^(?:[^\x00]{2,}?(?=(\{\{|\\\{\{|\\\\\{\{|$)))/,/^(?:[\s\S]*?--\}\})/,/^(?:\()/,/^(?:\))/,/^(?:\{\{(~)?>)/,/^(?:\{\{(~)?#)/,/^(?:\{\{(~)?\/)/,/^(?:\{\{(~)?\^)/,/^(?:\{\{(~)?\s*else\b)/,/^(?:\{\{(~)?\{)/,/^(?:\{\{(~)?&)/,/^(?:\{\{!--)/,/^(?:\{\{![\s\S]*?\}\})/,/^(?:\{\{(~)?)/,/^(?:=)/,/^(?:\.\.)/,/^(?:\.(?=([=~}\s\/.)])))/,/^(?:[\/.])/,/^(?:\s+)/,/^(?:\}(~)?\}\})/,/^(?:(~)?\}\})/,/^(?:"(\\["]|[^"])*")/,/^(?:'(\\[']|[^'])*')/,/^(?:@)/,/^(?:true(?=([~}\s)])))/,/^(?:false(?=([~}\s)])))/,/^(?:-?[0-9]+(?=([~}\s)])))/,/^(?:([^\s!"#%-,\.\/;->@\[-\^`\{-~]+(?=([=~}\s\/.)]))))/,/^(?:\[[^\]]*\])/,/^(?:.)/,/^(?:$)/];
  lexer.conditions = {"mu":{"rules":[4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32],"inclusive":false},"emu":{"rules":[2],"inclusive":false},"com":{"rules":[3],"inclusive":false},"INITIAL":{"rules":[0,1,32],"inclusive":true}};
  return lexer;})()
  parser.lexer = lexer;
  function Parser () {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[47010,47027],"range":[46991,47027],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":87,"label":"dijon"}'); this.yy = {}; }Parser.prototype = parser;parser.Parser = Parser;
  return new Parser;
  })();__exports__ = handlebars;
  /* jshint ignore:end */
  return __exports__;
})();

// handlebars/compiler/base.js
var __module8__ = (function(__dependency1__, __dependency2__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[47279,47676],"range":[47236,47676],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":91,"label":"dijon"}');
  "use strict";
  var __exports__ = {};
  var parser = __dependency1__;
  var AST = __dependency2__;

  __exports__.parser = parser;

  function parse(input) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[47439,47622],"range":[47417,47622],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":90,"label":"dijon"}');
    // Just return if an already-compile AST was passed in.
    if(input.constructor === AST.ProgramNode) { return input; }

    parser.yy = AST;
    return parser.parse(input);
  }

  __exports__.parse = parse;
  return __exports__;
})(__module9__, __module7__);

// handlebars/compiler/compiler.js
var __module10__ = (function(__dependency1__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[47787,60390],"range":[47761,60390],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":125,"label":"dijon"}');
  "use strict";
  var __exports__ = {};
  var Exception = __dependency1__;

  function Compiler() {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[47887,47889],"range":[47867,47889],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":92,"label":"dijon"}');}

  __exports__.Compiler = Compiler;// the foundHelper register will disambiguate helper lookup from finding a
  // function in a context. This is necessary for mustache compatibility, which
  // requires that context functions in blocks are evaluated by blockHelperMissing,
  // and then proceed as if the resulting value was provided to blockHelperMissing.

  Compiler.prototype = {
    compiler: Compiler,

    disassemble: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[48327,48996],"range":[48316,48996],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":93,"label":"dijon"}');
      var opcodes = this.opcodes, opcode, out = [], params, param;

      for (var i=0, l=opcodes.length; i<l; i++) {
        opcode = opcodes[i];

        if (opcode.opcode === 'DECLARE') {
          out.push("DECLARE " + opcode.name + "=" + opcode.value);
        } else {
          params = [];
          for (var j=0; j<opcode.args.length; j++) {
            param = opcode.args[j];
            if (typeof param === "string") {
              param = "\"" + param.replace("\n", "\\n") + "\"";
            }
            params.push(param);
          }
          out.push(opcode.opcode + " " + params.join(" "));
        }
      }

      return out.join("\n");
    },

    equals: function(other) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[49027,49830],"range":[49011,49830],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":94,"label":"dijon"}');
      var len = this.opcodes.length;
      if (other.opcodes.length !== len) {
        return false;
      }

      for (var i = 0; i < len; i++) {
        var opcode = this.opcodes[i],
            otherOpcode = other.opcodes[i];
        if (opcode.opcode !== otherOpcode.opcode || opcode.args.length !== otherOpcode.args.length) {
          return false;
        }
        for (var j = 0; j < opcode.args.length; j++) {
          if (opcode.args[j] !== otherOpcode.args[j]) {
            return false;
          }
        }
      }

      len = this.children.length;
      if (other.children.length !== len) {
        return false;
      }
      for (i = 0; i < len; i++) {
        if (!this.children[i].equals(other.children[i])) {
          return false;
        }
      }

      return true;
    },

    guid: 0,

    compile: function(program, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[49887,50536],"range":[49860,50536],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":95,"label":"dijon"}');
      this.opcodes = [];
      this.children = [];
      this.depths = {list: []};
      this.options = options;

      // These changes will propagate to the other compiler components
      var knownHelpers = this.options.knownHelpers;
      this.options.knownHelpers = {
        'helperMissing': true,
        'blockHelperMissing': true,
        'each': true,
        'if': true,
        'unless': true,
        'with': true,
        'log': true
      };
      if (knownHelpers) {
        for (var name in knownHelpers) {
          this.options.knownHelpers[name] = knownHelpers[name];
        }
      }

      return this.accept(program);
    },

    accept: function(node) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[50566,50805],"range":[50551,50805],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":96,"label":"dijon"}');
      var strip = node.strip || {},
          ret;
      if (strip.left) {
        this.opcode('strip');
      }

      ret = this[node.type](node);

      if (strip.right) {
        this.opcode('strip');
      }

      return ret;
    },

    program: function(program) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[50839,51134],"range":[50821,51134],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":98,"label":"dijon"}');
      var statements = program.statements;

      for(var i=0, l=statements.length; i<l; i++) {
        this.accept(statements[i]);
      }
      this.isSimple = l === 1;

      this.depths.list = this.depths.list.sort(function(a, b) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[51075,51106],"range":[51060,51106],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":97,"label":"dijon"}');
        return a - b;
      });

      return this;
    },

    compileProgram: function(program) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[51175,51598],"range":[51157,51598],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":99,"label":"dijon"}');
      var result = new this.compiler().compile(program, this.options);
      var guid = this.guid++, depth;

      this.usePartial = this.usePartial || result.usePartial;

      this.children[guid] = result;

      for(var i=0, l=result.depths.list.length; i<l; i++) {
        depth = result.depths.list[i];

        if(depth < 2) { continue; }
        else { this.addDepth(depth - 1); }
      }

      return guid;
    },

    block: function(block) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[51628,52809],"range":[51612,52809],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":100,"label":"dijon"}');
      var mustache = block.mustache,
          program = block.program,
          inverse = block.inverse;

      if (program) {
        program = this.compileProgram(program);
      }

      if (inverse) {
        inverse = this.compileProgram(inverse);
      }

      var sexpr = mustache.sexpr;
      var type = this.classifySexpr(sexpr);

      if (type === "helper") {
        this.helperSexpr(sexpr, program, inverse);
      } else if (type === "simple") {
        this.simpleSexpr(sexpr);

        // now that the simple mustache is resolved, we need to
        // evaluate it by executing `blockHelperMissing`
        this.opcode('pushProgram', program);
        this.opcode('pushProgram', inverse);
        this.opcode('emptyHash');
        this.opcode('blockValue');
      } else {
        this.ambiguousSexpr(sexpr, program, inverse);

        // now that the simple mustache is resolved, we need to
        // evaluate it by executing `blockHelperMissing`
        this.opcode('pushProgram', program);
        this.opcode('pushProgram', inverse);
        this.opcode('emptyHash');
        this.opcode('ambiguousBlockValue');
      }

      this.opcode('append');
    },

    hash: function(hash) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[52837,53574],"range":[52822,53574],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":101,"label":"dijon"}');
      var pairs = hash.pairs, pair, val;

      this.opcode('pushHash');

      for(var i=0, l=pairs.length; i<l; i++) {
        pair = pairs[i];
        val  = pair[1];

        if (this.options.stringParams) {
          if(val.depth) {
            this.addDepth(val.depth);
          }
          this.opcode('getContext', val.depth || 0);
          this.opcode('pushStringParam', val.stringModeValue, val.type);

          if (val.type === 'sexpr') {
            // Subexpressions get evaluated and passed in
            // in string params mode.
            this.sexpr(val);
          }
        } else {
          this.accept(val);
        }

        this.opcode('assignToHash', pair[0]);
      }
      this.opcode('popHash');
    },

    partial: function(partial) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[53608,53899],"range":[53590,53899],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":102,"label":"dijon"}');
      var partialName = partial.partialName;
      this.usePartial = true;

      if(partial.context) {
        this.ID(partial.context);
      } else {
        this.opcode('push', 'depth0');
      }

      this.opcode('invokePartial', partialName.name);
      this.opcode('append');
    },

    content: function(content) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[53933,53992],"range":[53915,53992],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":103,"label":"dijon"}');
      this.opcode('appendContent', content.string);
    },

    mustache: function(mustache) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[54028,54217],"range":[54009,54217],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":104,"label":"dijon"}');
      this.sexpr(mustache.sexpr);

      if(mustache.escaped && !this.options.noEscape) {
        this.opcode('appendEscaped');
      } else {
        this.opcode('append');
      }
    },

    ambiguousSexpr: function(sexpr, program, inverse) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[54274,54577],"range":[54240,54577],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":105,"label":"dijon"}');
      var id = sexpr.id,
          name = id.parts[0],
          isBlock = program != null || inverse != null;

      this.opcode('getContext', id.depth);

      this.opcode('pushProgram', program);
      this.opcode('pushProgram', inverse);

      this.opcode('invokeAmbiguous', name, isBlock);
    },

    simpleSexpr: function(sexpr) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[54613,54976],"range":[54597,54976],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":106,"label":"dijon"}');
      var id = sexpr.id;

      if (id.type === 'DATA') {
        this.DATA(id);
      } else if (id.parts.length) {
        this.ID(id);
      } else {
        // Simplified ID for `this`
        this.addDepth(id.depth);
        this.opcode('getContext', id.depth);
        this.opcode('pushContext');
      }

      this.opcode('resolvePossibleLambda');
    },

    helperSexpr: function(sexpr, program, inverse) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[55030,55508],"range":[54996,55508],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":107,"label":"dijon"}');
      var params = this.setupFullMustacheParams(sexpr, program, inverse),
          name = sexpr.id.parts[0];

      if (this.options.knownHelpers[name]) {
        this.opcode('invokeKnownHelper', params.length, name);
      } else if (this.options.knownHelpersOnly) {
        throw new Exception("You specified knownHelpersOnly, but used the unknown helper " + name, sexpr);
      } else {
        this.opcode('invokeHelper', params.length, name, sexpr.isRoot);
      }
    },

    sexpr: function(sexpr) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[55538,55784],"range":[55522,55784],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":108,"label":"dijon"}');
      var type = this.classifySexpr(sexpr);

      if (type === "simple") {
        this.simpleSexpr(sexpr);
      } else if (type === "helper") {
        this.helperSexpr(sexpr);
      } else {
        this.ambiguousSexpr(sexpr);
      }
    },

    ID: function(id) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[55808,56154],"range":[55795,56154],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":109,"label":"dijon"}');
      this.addDepth(id.depth);
      this.opcode('getContext', id.depth);

      var name = id.parts[0];
      if (!name) {
        this.opcode('pushContext');
      } else {
        this.opcode('lookupOnContext', id.parts[0]);
      }

      for(var i=1, l=id.parts.length; i<l; i++) {
        this.opcode('lookup', id.parts[i]);
      }
    },

    DATA: function(data) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56182,56536],"range":[56167,56536],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":110,"label":"dijon"}');
      this.options.data = true;
      if (data.id.isScoped || data.id.depth) {
        throw new Exception('Scoped data references are not supported: ' + data.original, data);
      }

      this.opcode('lookupData');
      var parts = data.id.parts;
      for(var i=0, l=parts.length; i<l; i++) {
        this.opcode('lookup', parts[i]);
      }
    },

    STRING: function(string) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56568,56623],"range":[56551,56623],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":111,"label":"dijon"}');
      this.opcode('pushString', string.string);
    },

    INTEGER: function(integer) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56657,56715],"range":[56639,56715],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":112,"label":"dijon"}');
      this.opcode('pushLiteral', integer.integer);
    },

    BOOLEAN: function(bool) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56746,56798],"range":[56731,56798],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":113,"label":"dijon"}');
      this.opcode('pushLiteral', bool.bool);
    },

    comment: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56825,56827],"range":[56814,56827],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":114,"label":"dijon"}');},

    // HELPERS
    opcode: function(name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56872,56957],"range":[56857,56957],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":115,"label":"dijon"}');
      this.opcodes.push({ opcode: name, args: [].slice.call(arguments, 1) });
    },

    declare: function(name, value) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[56995,57076],"range":[56973,57076],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":116,"label":"dijon"}');
      this.opcodes.push({ opcode: 'DECLARE', name: name, value: value });
    },

    addDepth: function(depth) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[57109,57264],"range":[57093,57264],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":117,"label":"dijon"}');
      if(depth === 0) { return; }

      if(!this.depths[depth]) {
        this.depths[depth] = true;
        this.depths.list.push(depth);
      }
    },

    classifySexpr: function(sexpr) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[57302,57861],"range":[57286,57861],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":118,"label":"dijon"}');
      var isHelper   = sexpr.isHelper;
      var isEligible = sexpr.eligibleHelper;
      var options    = this.options;

      // if ambiguous, we can possibly resolve the ambiguity now
      if (isEligible && !isHelper) {
        var name = sexpr.id.parts[0];

        if (options.knownHelpers[name]) {
          isHelper = true;
        } else if (options.knownHelpersOnly) {
          isEligible = false;
        }
      }

      if (isHelper) { return "helper"; }
      else if (isEligible) { return "ambiguous"; }
      else { return "simple"; }
    },

    pushParams: function(params) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[57897,58491],"range":[57880,58491],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":119,"label":"dijon"}');
      var i = params.length, param;

      while(i--) {
        param = params[i];

        if(this.options.stringParams) {
          if(param.depth) {
            this.addDepth(param.depth);
          }

          this.opcode('getContext', param.depth || 0);
          this.opcode('pushStringParam', param.stringModeValue, param.type);

          if (param.type === 'sexpr') {
            // Subexpressions get evaluated and passed in
            // in string params mode.
            this.sexpr(param);
          }
        } else {
          this[param.type](param);
        }
      }
    },

    setupFullMustacheParams: function(sexpr, program, inverse) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[58557,58850],"range":[58523,58850],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":120,"label":"dijon"}');
      var params = sexpr.params;
      this.pushParams(params);

      this.opcode('pushProgram', program);
      this.opcode('pushProgram', inverse);

      if (sexpr.hash) {
        this.hash(sexpr.hash);
      } else {
        this.opcode('emptyHash');
      }

      return params;
    }
  };

  function precompile(input, options, env) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[58900,59392],"range":[58859,59392],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":121,"label":"dijon"}');
    if (input == null || (typeof input !== 'string' && input.constructor !== env.AST.ProgramNode)) {
      throw new Exception("You must pass a string or Handlebars AST to Handlebars.precompile. You passed " + input);
    }

    options = options || {};
    if (!('data' in options)) {
      options.data = true;
    }

    var ast = env.parse(input);
    var environment = new env.Compiler().compile(ast, options);
    return new env.JavaScriptCompiler().compile(environment, options);
  }

  __exports__.precompile = precompile;function compile(input, options, env) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[59470,60332],"range":[59432,60332],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":124,"label":"dijon"}');
    if (input == null || (typeof input !== 'string' && input.constructor !== env.AST.ProgramNode)) {
      throw new Exception("You must pass a string or Handlebars AST to Handlebars.compile. You passed " + input);
    }

    options = options || {};

    if (!('data' in options)) {
      options.data = true;
    }

    var compiled;

    function compileInput() {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[59837,60087],"range":[59813,60087],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":122,"label":"dijon"}');
      var ast = env.parse(input);
      var environment = new env.Compiler().compile(ast, options);
      var templateSpec = new env.JavaScriptCompiler().compile(environment, options, undefined, true);
      return env.template(templateSpec);
    }

    // Template is only compiled on first use and cached after that point.
    return function(context, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[60202,60327],"range":[60175,60327],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":123,"label":"dijon"}');
      if (!compiled) {
        compiled = compileInput();
      }
      return compiled.call(this, context, options);
    };
  }

  __exports__.compile = compile;
  return __exports__;
})(__module5__);

// handlebars/compiler/javascript-compiler.js
var __module11__ = (function(__dependency1__, __dependency2__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[60516,88448],"range":[60473,88448],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":186,"label":"dijon"}');
  "use strict";
  var __exports__;
  var COMPILER_REVISION = __dependency1__.COMPILER_REVISION;
  var REVISION_CHANGES = __dependency1__.REVISION_CHANGES;
  var log = __dependency1__.log;
  var Exception = __dependency2__;

  function Literal(value) {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[60768,60797],"range":[60744,60797],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":126,"label":"dijon"}');
    this.value = value;
  }

  function JavaScriptCompiler() {instrumentation_log('{"type":"FunctionDeclaration","bodyRange":[60831,60833],"range":[60801,60833],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":127,"label":"dijon"}');}

  JavaScriptCompiler.prototype = {
    // PUBLIC API: You can override these methods in a subclass to provide
    // alternative compiled forms for name lookup and buffering semantics
    nameLookup: function(parent, name /* , type*/) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[61070,61551],"range":[61035,61551],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":128,"label":"dijon"}');
      var wrap,
          ret;
      if (parent.indexOf('depth') === 0) {
        wrap = true;
      }

      if (/^[0-9]+$/.test(name)) {
        ret = parent + "[" + name + "]";
      } else if (JavaScriptCompiler.isValidJavaScriptVariableName(name)) {
        ret = parent + "." + name;
      }
      else {
        ret = parent + "['" + name + "']";
      }

      if (wrap) {
        return '(' + parent + ' && ' + ret + ')';
      } else {
        return ret;
      }
    },

    compilerInfo: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[61583,61748],"range":[61572,61748],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":129,"label":"dijon"}');
      var revision = COMPILER_REVISION,
          versions = REVISION_CHANGES[revision];
      return "this.compilerInfo = ["+revision+",'"+versions+"'];\n";
    },

    appendToBuffer: function(string) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[61788,62056],"range":[61771,62056],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":131,"label":"dijon"}');
      if (this.environment.isSimple) {
        return "return " + string + ";";
      } else {
        return {
          appendToBuffer: true,
          content: string,
          toString: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[61992,62031],"range":[61981,62031],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":130,"label":"dijon"}'); return "buffer += " + string + ";"; }
        };
      }
    },

    initializeBuffer: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[62092,62135],"range":[62081,62135],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":132,"label":"dijon"}');
      return this.quotedString("");
    },

    namespace: "Handlebars",
    // END PUBLIC API

    compile: function(environment, options, context, asObject) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[62253,63641],"range":[62203,63641],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":133,"label":"dijon"}');
      this.environment = environment;
      this.options = options || {};

      log('debug', this.environment.disassemble() + "\n\n");

      this.name = this.environment.name;
      this.isChild = !!context;
      this.context = context || {
        programs: [],
        environments: [],
        aliases: { }
      };

      this.preamble();

      this.stackSlot = 0;
      this.stackVars = [];
      this.registers = { list: [] };
      this.hashes = [];
      this.compileStack = [];
      this.inlineStack = [];

      this.compileChildren(environment, options);

      var opcodes = environment.opcodes, opcode;

      this.i = 0;

      for(var l=opcodes.length; this.i<l; this.i++) {
        opcode = opcodes[this.i];

        if(opcode.opcode === 'DECLARE') {
          this[opcode.name] = opcode.value;
        } else {
          this[opcode.opcode].apply(this, opcode.args);
        }

        // Reset the stripNext flag if it was not set by this operation.
        if (opcode.opcode !== this.stripNext) {
          this.stripNext = false;
        }
      }

      // Flush any trailing content that might be pending.
      this.pushSource('');

      if (this.stackSlot || this.inlineStack.length || this.compileStack.length) {
        throw new Exception('Compile completed with content left on stack');
      }

      return this.createFunctionContext(asObject);
    },

    preamble: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[63669,64444],"range":[63658,64444],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":134,"label":"dijon"}');
      var out = [];

      if (!this.isChild) {
        var namespace = this.namespace;

        var copies = "helpers = this.merge(helpers, " + namespace + ".helpers);";
        if (this.environment.usePartial) { copies = copies + " partials = this.merge(partials, " + namespace + ".partials);"; }
        if (this.options.data) { copies = copies + " data = data || {};"; }
        out.push(copies);
      } else {
        out.push('');
      }

      if (!this.environment.isSimple) {
        out.push(", buffer = " + this.initializeBuffer());
      } else {
        out.push("");
      }

      // track the last context pushed into place to allow skipping the
      // getContext opcode when it would be a noop
      this.lastContext = 0;
      this.source = out;
    },

    createFunctionContext: function(asObject) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[64493,66077],"range":[64474,66077],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":135,"label":"dijon"}');
      var locals = this.stackVars.concat(this.registers.list);

      if(locals.length > 0) {
        this.source[1] = this.source[1] + ", " + locals.join(", ");
      }

      // Generate minimizer alias mappings
      if (!this.isChild) {
        for (var alias in this.context.aliases) {
          if (this.context.aliases.hasOwnProperty(alias)) {
            this.source[1] = this.source[1] + ', ' + alias + '=' + this.context.aliases[alias];
          }
        }
      }

      if (this.source[1]) {
        this.source[1] = "var " + this.source[1].substring(2) + ";";
      }

      // Merge children
      if (!this.isChild) {
        this.source[1] += '\n' + this.context.programs.join('\n') + '\n';
      }

      if (!this.environment.isSimple) {
        this.pushSource("return buffer;");
      }

      var params = this.isChild ? ["depth0", "data"] : ["Handlebars", "depth0", "helpers", "partials", "data"];

      for(var i=0, l=this.environment.depths.list.length; i<l; i++) {
        params.push("depth" + this.environment.depths.list[i]);
      }

      // Perform a second pass over the output to merge content when possible
      var source = this.mergeSource();

      if (!this.isChild) {
        source = this.compilerInfo()+source;
      }

      if (asObject) {
        params.push(source);

        return Function.apply(this, params);
      } else {
        var functionSource = 'function ' + (this.name || '') + '(' + params.join(',') + ') {\n  ' + source + '}';
        log('debug', functionSource + "\n\n");
        return functionSource;
      }
    },
    mergeSource: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[66107,66813],"range":[66096,66813],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":136,"label":"dijon"}');
      // WARN: We are not handling the case where buffer is still populated as the source should
      // not have buffer append operations as their final action.
      var source = '',
          buffer;
      for (var i = 0, len = this.source.length; i < len; i++) {
        var line = this.source[i];
        if (line.appendToBuffer) {
          if (buffer) {
            buffer = buffer + '\n    + ' + line.content;
          } else {
            buffer = line.content;
          }
        } else {
          if (buffer) {
            source += 'buffer += ' + buffer + ';\n  ';
            buffer = undefined;
          }
          source += line + '\n  ';
        }
      }
      return source;
    },

    // [blockValue]
    //
    // On stack, before: hash, inverse, program, value
    // On stack, after: return value of blockHelperMissing
    //
    // The purpose of this opcode is to take a block of the form
    // `{{#foo}}...{{/foo}}`, resolve the value of `foo`, and
    // replace it on the stack with the result of properly
    // invoking blockHelperMissing.
    blockValue: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[67213,67527],"range":[67202,67527],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":138,"label":"dijon"}');
      this.context.aliases.blockHelperMissing = 'helpers.blockHelperMissing';

      var params = ["depth0"];
      this.setupParams(0, params);

      this.replaceStack(function(current) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[67403,67519],"range":[67385,67519],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":137,"label":"dijon"}');
        params.splice(1, 0, current);
        return "blockHelperMissing.call(" + params.join(", ") + ")";
      });
    },

    // [ambiguousBlockValue]
    //
    // On stack, before: hash, inverse, program, value
    // Compiler value, before: lastHelper=value of last found helper, if any
    // On stack, after, if no lastHelper: same as [blockValue]
    // On stack, after, if lastHelper: value
    ambiguousBlockValue: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[67842,68199],"range":[67831,68199],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":139,"label":"dijon"}');
      this.context.aliases.blockHelperMissing = 'helpers.blockHelperMissing';

      var params = ["depth0"];
      this.setupParams(0, params);

      var current = this.topStack();
      params.splice(1, 0, current);

      this.pushSource("if (!" + this.lastHelper + ") { " + current + " = blockHelperMissing.call(" + params.join(", ") + "); }");
    },

    // [appendContent]
    //
    // On stack, before: ...
    // On stack, after: ...
    //
    // Appends the string value of `content` to the current buffer
    appendContent: function(content) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[68400,68618],"range":[68382,68618],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":140,"label":"dijon"}');
      if (this.pendingContent) {
        content = this.pendingContent + content;
      }
      if (this.stripNext) {
        content = content.replace(/^\s+/, '');
      }

      this.pendingContent = content;
    },

    // [strip]
    //
    // On stack, before: ...
    // On stack, after: ...
    //
    // Removes any trailing whitespace from the prior content node and flags
    // the next operation for stripping if it is a content node.
    strip: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[68871,69022],"range":[68860,69022],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":141,"label":"dijon"}');
      if (this.pendingContent) {
        this.pendingContent = this.pendingContent.replace(/\s+$/, '');
      }
      this.stripNext = 'strip';
    },

    // [append]
    //
    // On stack, before: value, ...
    // On stack, after: ...
    //
    // Coerces `value` to a String and appends it to the current buffer.
    //
    // If `value` is truthy, or 0, it is coerced into a string and appended
    // Otherwise, the empty string is appended
    append: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[69345,69752],"range":[69334,69752],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":142,"label":"dijon"}');
      // Force anything that is inlined onto the stack so we don't have duplication
      // when we examine local
      this.flushInline();
      var local = this.popStack();
      this.pushSource("if(" + local + " || " + local + " === 0) { " + this.appendToBuffer(local) + " }");
      if (this.environment.isSimple) {
        this.pushSource("else { " + this.appendToBuffer("''") + " }");
      }
    },

    // [appendEscaped]
    //
    // On stack, before: value, ...
    // On stack, after: ...
    //
    // Escape `value` and append it to the buffer
    appendEscaped: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[69936,70104],"range":[69925,70104],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":143,"label":"dijon"}');
      this.context.aliases.escapeExpression = 'this.escapeExpression';

      this.pushSource(this.appendToBuffer("escapeExpression(" + this.popStack() + ")"));
    },

    // [getContext]
    //
    // On stack, before: ...
    // On stack, after: ...
    // Compiler value, after: lastContext=depth
    //
    // Set the value of the `lastContext` compiler value to the depth
    getContext: function(depth) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[70348,70436],"range":[70332,70436],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":144,"label":"dijon"}');
      if(this.lastContext !== depth) {
        this.lastContext = depth;
      }
    },

    // [lookupOnContext]
    //
    // On stack, before: ...
    // On stack, after: currentContext[name], ...
    //
    // Looks up the value of `name` on the current context and pushes
    // it onto the stack.
    lookupOnContext: function(name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[70689,70775],"range":[70674,70775],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":145,"label":"dijon"}');
      this.push(this.nameLookup('depth' + this.lastContext, name, 'context'));
    },

    // [pushContext]
    //
    // On stack, before: ...
    // On stack, after: currentContext, ...
    //
    // Pushes the value of the current context onto the stack.
    pushContext: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[70977,71041],"range":[70966,71041],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":146,"label":"dijon"}');
      this.pushStackLiteral('depth' + this.lastContext);
    },

    // [resolvePossibleLambda]
    //
    // On stack, before: value, ...
    // On stack, after: resolved value, ...
    //
    // If the `value` is a lambda, replace it on the stack by
    // the return value of the lambda
    resolvePossibleLambda: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[71307,71528],"range":[71296,71528],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":148,"label":"dijon"}');
      this.context.aliases.functionType = '"function"';

      this.replaceStack(function(current) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[71408,71520],"range":[71390,71520],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":147,"label":"dijon"}');
        return "typeof " + current + " === functionType ? " + current + ".apply(depth0) : " + current;
      });
    },

    // [lookup]
    //
    // On stack, before: value, ...
    // On stack, after: value[name], ...
    //
    // Replace the value on the stack with the result of looking
    // up `name` on `value`
    lookup: function(name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[71758,71950],"range":[71743,71950],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":150,"label":"dijon"}');
      this.replaceStack(function(current) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[71802,71942],"range":[71784,71942],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":149,"label":"dijon"}');
        return current + " == null || " + current + " === false ? " + current + " : " + this.nameLookup(current, name, 'context');
      });
    },

    // [lookupData]
    //
    // On stack, before: ...
    // On stack, after: data, ...
    //
    // Push the data lookup operator
    lookupData: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[72114,72158],"range":[72103,72158],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":151,"label":"dijon"}');
      this.pushStackLiteral('data');
    },

    // [pushStringParam]
    //
    // On stack, before: ...
    // On stack, after: string, currentContext, ...
    //
    // This opcode is designed for use in string mode, which
    // provides the string value of a parameter along with its
    // depth rather than resolving it immediately.
    pushStringParam: function(string, type) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[72500,72873],"range":[72477,72873],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":152,"label":"dijon"}');
      this.pushStackLiteral('depth' + this.lastContext);

      this.pushString(type);

      // If it's a subexpression, the string result
      // will be pushed after this opcode.
      if (type !== 'sexpr') {
        if (typeof string === 'string') {
          this.pushString(string);
        } else {
          this.pushStackLiteral(string);
        }
      }
    },

    emptyHash: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[72902,73071],"range":[72891,73071],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":153,"label":"dijon"}');
      this.pushStackLiteral('{}');

      if (this.options.stringParams) {
        this.push('{}'); // hashContexts
        this.push('{}'); // hashTypes
      }
    },
    pushHash: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[73098,73230],"range":[73087,73230],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":154,"label":"dijon"}');
      if (this.hash) {
        this.hashes.push(this.hash);
      }
      this.hash = {values: [], types: [], contexts: []};
    },
    popHash: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[73256,73554],"range":[73245,73554],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":155,"label":"dijon"}');
      var hash = this.hash;
      this.hash = this.hashes.pop();

      if (this.options.stringParams) {
        this.push('{' + hash.contexts.join(',') + '}');
        this.push('{' + hash.types.join(',') + '}');
      }

      this.push('{\n    ' + hash.values.join(',\n    ') + '\n  }');
    },

    // [pushString]
    //
    // On stack, before: ...
    // On stack, after: quotedString(string), ...
    //
    // Push a quoted version of `string` onto the stack
    pushString: function(string) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[73759,73822],"range":[73742,73822],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":156,"label":"dijon"}');
      this.pushStackLiteral(this.quotedString(string));
    },

    // [push]
    //
    // On stack, before: ...
    // On stack, after: expr, ...
    //
    // Push an expression onto the stack
    push: function(expr) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[73982,74043],"range":[73967,74043],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":157,"label":"dijon"}');
      this.inlineStack.push(expr);
      return expr;
    },

    // [pushLiteral]
    //
    // On stack, before: ...
    // On stack, after: value, ...
    //
    // Pushes a value onto the stack. This operation prevents
    // the compiler from creating a temporary variable to hold
    // it.
    pushLiteral: function(value) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[74314,74357],"range":[74298,74357],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":158,"label":"dijon"}');
      this.pushStackLiteral(value);
    },

    // [pushProgram]
    //
    // On stack, before: ...
    // On stack, after: program(guid), ...
    //
    // Push a program expression onto the stack. This takes
    // a compile-time guid and converts it into a runtime-accessible
    // expression.
    pushProgram: function(guid) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[74647,74801],"range":[74632,74801],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":159,"label":"dijon"}');
      if (guid != null) {
        this.pushStackLiteral(this.programExpression(guid));
      } else {
        this.pushStackLiteral(null);
      }
    },

    // [invokeHelper]
    //
    // On stack, before: hash, inverse, program, params..., ...
    // On stack, after: result of helper invocation
    //
    // Pops off the helper's parameters, invokes the helper,
    // and pushes the helper's return value onto the stack.
    //
    // If the helper is not found, `helperMissing` is called.
    invokeHelper: function(paramSize, name, isRoot) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[75198,76155],"range":[75164,76155],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":160,"label":"dijon"}');
      this.context.aliases.helperMissing = 'helpers.helperMissing';
      this.useRegister('helper');

      var helper = this.lastHelper = this.setupHelper(paramSize, name, true);
      var nonHelper = this.nameLookup('depth' + this.lastContext, name, 'context');

      var lookup = 'helper = ' + helper.name + ' || ' + nonHelper;
      if (helper.paramsInit) {
        lookup += ',' + helper.paramsInit;
      }

      this.push(
        '('
          + lookup
          + ',helper '
            + '? helper.call(' + helper.callParams + ') '
            + ': helperMissing.call(' + helper.helperMissingParams + '))');

      // Always flush subexpressions. This is both to prevent the compounding size issue that
      // occurs when the code has to be duplicated for inlining and also to prevent errors
      // due to the incorrect options object being passed due to the shared register.
      if (!isRoot) {
        this.flushInline();
      }
    },

    // [invokeKnownHelper]
    //
    // On stack, before: hash, inverse, program, params..., ...
    // On stack, after: result of helper invocation
    //
    // This operation is used when the helper is known to exist,
    // so a `helperMissing` fallback is not required.
    invokeKnownHelper: function(paramSize, name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[76483,76611],"range":[76457,76611],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":161,"label":"dijon"}');
      var helper = this.setupHelper(paramSize, name);
      this.push(helper.name + ".call(" + helper.callParams + ")");
    },

    // [invokeAmbiguous]
    //
    // On stack, before: hash, inverse, program, params..., ...
    // On stack, after: result of disambiguation
    //
    // This operation is used when an expression like `{{foo}}`
    // is provided, but we don't know at compile-time whether it
    // is a helper or a path.
    //
    // This operation emits more code than the other options,
    // and can be avoided by passing the `knownHelpers` and
    // `knownHelpersOnly` flags at compile-time.
    invokeAmbiguous: function(name, helperCall) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[77151,77914],"range":[77124,77914],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":162,"label":"dijon"}');
      this.context.aliases.functionType = '"function"';
      this.useRegister('helper');

      this.emptyHash();
      var helper = this.setupHelper(0, name, helperCall);

      var helperName = this.lastHelper = this.nameLookup('helpers', name, 'helper');

      var nonHelper = this.nameLookup('depth' + this.lastContext, name, 'context');
      var nextStack = this.nextStack();

      if (helper.paramsInit) {
        this.pushSource(helper.paramsInit);
      }
      this.pushSource('if (helper = ' + helperName + ') { ' + nextStack + ' = helper.call(' + helper.callParams + '); }');
      this.pushSource('else { helper = ' + nonHelper + '; ' + nextStack + ' = typeof helper === functionType ? helper.call(' + helper.callParams + ') : helper; }');
    },

    // [invokePartial]
    //
    // On stack, before: context, ...
    // On stack after: result of partial invocation
    //
    // This operation pops off a context, invokes a partial with that context,
    // and pushes the result of the invocation back.
    invokePartial: function(name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[78210,78520],"range":[78195,78520],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":163,"label":"dijon"}');
      var params = [this.nameLookup('partials', name, 'partial'), "'" + name + "'", this.popStack(), "helpers", "partials"];

      if (this.options.data) {
        params.push("data");
      }

      this.context.aliases.self = "this";
      this.push("self.invokePartial(" + params.join(", ") + ")");
    },

    // [assignToHash]
    //
    // On stack, before: value, hash, ...
    // On stack, after: hash, ...
    //
    // Pops a value and hash off the stack, assigns `hash[key] = value`
    // and pushes the hash back onto the stack.
    assignToHash: function(key) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[78787,79229],"range":[78773,79229],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":164,"label":"dijon"}');
      var value = this.popStack(),
          context,
          type;

      if (this.options.stringParams) {
        type = this.popStack();
        context = this.popStack();
      }

      var hash = this.hash;
      if (context) {
        hash.contexts.push("'" + key + "': " + context);
      }
      if (type) {
        hash.types.push("'" + key + "': " + type);
      }
      hash.values.push("'" + key + "': (" + value + ")");
    },

    // HELPERS

    compiler: JavaScriptCompiler,

    compileChildren: function(environment, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[79335,80084],"range":[79304,80084],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":165,"label":"dijon"}');
      var children = environment.children, child, compiler;

      for(var i=0, l=children.length; i<l; i++) {
        child = children[i];
        compiler = new this.compiler();

        var index = this.matchExistingProgram(child);

        if (index == null) {
          this.context.programs.push('');     // Placeholder to prevent name conflicts for nested children
          index = this.context.programs.length;
          child.index = index;
          child.name = 'program' + index;
          this.context.programs[index] = compiler.compile(child, options, this.context);
          this.context.environments[index] = child;
        } else {
          child.index = index;
          child.name = 'program' + index;
        }
      }
    },
    matchExistingProgram: function(child) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[80128,80363],"range":[80112,80363],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":166,"label":"dijon"}');
      for (var i = 0, len = this.context.environments.length; i < len; i++) {
        var environment = this.context.environments[i];
        if (environment && environment.equals(child)) {
          return i;
        }
      }
    },

    programExpression: function(guid) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[80404,80994],"range":[80389,80994],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":167,"label":"dijon"}');
      this.context.aliases.self = "this";

      if(guid == null) {
        return "self.noop";
      }

      var child = this.environment.children[guid],
          depths = child.depths.list, depth;

      var programParams = [child.index, child.name, "data"];

      for(var i=0, l = depths.length; i<l; i++) {
        depth = depths[i];

        if(depth === 1) { programParams.push("depth0"); }
        else { programParams.push("depth" + (depth - 1)); }
      }

      return (depths.length === 0 ? "self.program(" : "self.programWithDepth(") + programParams.join(", ") + ")";
    },

    register: function(name, val) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[81031,81117],"range":[81011,81117],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":168,"label":"dijon"}');
      this.useRegister(name);
      this.pushSource(name + " = " + val + ";");
    },

    useRegister: function(name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[81152,81278],"range":[81137,81278],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":169,"label":"dijon"}');
      if(!this.registers[name]) {
        this.registers[name] = true;
        this.registers.list.push(name);
      }
    },

    pushStackLiteral: function(item) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[81318,81368],"range":[81303,81368],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":170,"label":"dijon"}');
      return this.push(new Literal(item));
    },

    pushSource: function(source) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[81404,81643],"range":[81387,81643],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":171,"label":"dijon"}');
      if (this.pendingContent) {
        this.source.push(this.appendToBuffer(this.quotedString(this.pendingContent)));
        this.pendingContent = undefined;
      }

      if (source) {
        this.source.push(source);
      }
    },

    pushStack: function(item) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[81676,81882],"range":[81661,81882],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":172,"label":"dijon"}');
      this.flushInline();

      var stack = this.incrStack();
      if (item) {
        this.pushSource(stack + " = " + item + ";");
      }
      this.compileStack.push(stack);
      return stack;
    },

    replaceStack: function(callback) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[81922,83277],"range":[81903,83277],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":173,"label":"dijon"}');
      var prefix = '',
          inline = this.isInline(),
          stack,
          createdStack,
          usedLiteral;

      // If we are currently inline then we want to merge the inline statement into the
      // replacement statement via ','
      if (inline) {
        var top = this.popStack(true);

        if (top instanceof Literal) {
          // Literals do not need to be inlined
          stack = top.value;
          usedLiteral = true;
        } else {
          // Get or create the current stack name for use by the inline
          createdStack = !this.stackSlot;
          var name = !createdStack ? this.topStackName() : this.incrStack();

          prefix = '(' + this.push(name) + ' = ' + top + '),';
          stack = this.topStack();
        }
      } else {
        stack = this.topStack();
      }

      var item = callback.call(this, stack);

      if (inline) {
        if (!usedLiteral) {
          this.popStack();
        }
        if (createdStack) {
          this.stackSlot--;
        }
        this.push('(' + prefix + item + ')');
      } else {
        // Prevent modification of the context depth variable. Through replaceStack
        if (!/^stack/.test(stack)) {
          stack = this.nextStack();
        }

        this.pushSource(stack + " = (" + prefix + item + ");");
      }
      return stack;
    },

    nextStack: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[83306,83344],"range":[83295,83344],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":174,"label":"dijon"}');
      return this.pushStack();
    },

    incrStack: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[83373,83538],"range":[83362,83538],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":175,"label":"dijon"}');
      this.stackSlot++;
      if(this.stackSlot > this.stackVars.length) { this.stackVars.push("stack" + this.stackSlot); }
      return this.topStackName();
    },
    topStackName: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[83569,83615],"range":[83558,83615],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":176,"label":"dijon"}');
      return "stack" + this.stackSlot;
    },
    flushInline: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[83645,84030],"range":[83634,84030],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":177,"label":"dijon"}');
      var inlineStack = this.inlineStack;
      if (inlineStack.length) {
        this.inlineStack = [];
        for (var i = 0, len = inlineStack.length; i < len; i++) {
          var entry = inlineStack[i];
          if (entry instanceof Literal) {
            this.compileStack.push(entry);
          } else {
            this.pushStack(entry);
          }
        }
      }
    },
    isInline: function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[84057,84102],"range":[84046,84102],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":178,"label":"dijon"}');
      return this.inlineStack.length;
    },

    popStack: function(wrapped) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[84137,84535],"range":[84119,84535],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":179,"label":"dijon"}');
      var inline = this.isInline(),
          item = (inline ? this.inlineStack : this.compileStack).pop();

      if (!wrapped && (item instanceof Literal)) {
        return item.value;
      } else {
        if (!inline) {
          if (!this.stackSlot) {
            throw new Exception('Invalid stack pop');
          }
          this.stackSlot--;
        }
        return item;
      }
    },

    topStack: function(wrapped) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[84570,84818],"range":[84552,84818],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":180,"label":"dijon"}');
      var stack = (this.isInline() ? this.inlineStack : this.compileStack),
          item = stack[stack.length - 1];

      if (!wrapped && (item instanceof Literal)) {
        return item.value;
      } else {
        return item;
      }
    },

    quotedString: function(str) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[84853,85122],"range":[84839,85122],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":181,"label":"dijon"}');
      return '"' + str
        .replace(/\\/g, '\\\\')
        .replace(/"/g, '\\"')
        .replace(/\n/g, '\\n')
        .replace(/\r/g, '\\r')
        .replace(/\u2028/g, '\\u2028')   // Per Ecma-262 7.3 + 7.8.4
        .replace(/\u2029/g, '\\u2029') + '"';
    },

    setupHelper: function(paramSize, name, missingParams) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[85183,85630],"range":[85142,85630],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":182,"label":"dijon"}');
      var params = [],
          paramsInit = this.setupParams(paramSize, params, missingParams);
      var foundHelper = this.nameLookup('helpers', name, 'helper');

      return {
        params: params,
        paramsInit: paramsInit,
        name: foundHelper,
        callParams: ["depth0"].concat(params).join(", "),
        helperMissingParams: missingParams && ["depth0", this.quotedString(name)].concat(params).join(", ")
      };
    },

    setupOptions: function(paramSize, params) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[85679,87017],"range":[85651,87017],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":183,"label":"dijon"}');
      var options = [], contexts = [], types = [], param, inverse, program;

      options.push("hash:" + this.popStack());

      if (this.options.stringParams) {
        options.push("hashTypes:" + this.popStack());
        options.push("hashContexts:" + this.popStack());
      }

      inverse = this.popStack();
      program = this.popStack();

      // Avoid setting fn and inverse if neither are set. This allows
      // helpers to do a check for `if (options.fn)`
      if (program || inverse) {
        if (!program) {
          this.context.aliases.self = "this";
          program = "self.noop";
        }

        if (!inverse) {
          this.context.aliases.self = "this";
          inverse = "self.noop";
        }

        options.push("inverse:" + inverse);
        options.push("fn:" + program);
      }

      for(var i=0; i<paramSize; i++) {
        param = this.popStack();
        params.push(param);

        if(this.options.stringParams) {
          types.push(this.popStack());
          contexts.push(this.popStack());
        }
      }

      if (this.options.stringParams) {
        options.push("contexts:[" + contexts.join(",") + "]");
        options.push("types:[" + types.join(",") + "]");
      }

      if(this.options.data) {
        options.push("data:data");
      }

      return options;
    },

    // the params and contexts arguments are passed in arrays
    // to fill in
    setupParams: function(paramSize, params, useRegister) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[87158,87449],"range":[87117,87449],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":184,"label":"dijon"}');
      var options = '{' + this.setupOptions(paramSize, params).join(',') + '}';

      if (useRegister) {
        this.useRegister('options');
        params.push('options');
        return 'options=' + options;
      } else {
        params.push(options);
        return '';
      }
    }
  };

  var reservedWords = (
    "break else new var" +
    " case finally return void" +
    " catch for switch while" +
    " continue function this with" +
    " default if throw" +
    " delete in try" +
    " do instanceof typeof" +
    " abstract enum int short" +
    " boolean export interface static" +
    " byte extends long super" +
    " char final native synchronized" +
    " class float package throws" +
    " const goto private transient" +
    " debugger implements protected volatile" +
    " double import public let yield"
  ).split(" ");

  var compilerWords = JavaScriptCompiler.RESERVED_WORDS = {};

  for(var i=0, l=reservedWords.length; i<l; i++) {
    compilerWords[reservedWords[i]] = true;
  }

  JavaScriptCompiler.isValidJavaScriptVariableName = function(name) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[88244,88386],"range":[88229,88386],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":185,"label":"dijon"}');
    if(!JavaScriptCompiler.RESERVED_WORDS[name] && /^[a-zA-Z_$][0-9a-zA-Z_$]*$/.test(name)) {
      return true;
    }
    return false;
  };

  __exports__ = JavaScriptCompiler;
  return __exports__;
})(__module2__, __module5__);

// handlebars.js
var __module0__ = (function(__dependency1__, __dependency2__, __dependency3__, __dependency4__, __dependency5__) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[88608,89568],"range":[88514,89568],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":190,"label":"dijon"}');
  "use strict";
  var __exports__;
  /*globals Handlebars: true */
  var Handlebars = __dependency1__;

  // Compiler imports
  var AST = __dependency2__;
  var Parser = __dependency3__.parser;
  var parse = __dependency3__.parse;
  var Compiler = __dependency4__.Compiler;
  var compile = __dependency4__.compile;
  var precompile = __dependency4__.precompile;
  var JavaScriptCompiler = __dependency5__;

  var _create = Handlebars.create;
  var create = function() {instrumentation_log('{"type":"FunctionExpression","bodyRange":[89078,89458],"range":[89067,89458],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":189,"label":"dijon"}');
    var hb = _create();

    hb.compile = function(input, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[89147,89196],"range":[89122,89196],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":187,"label":"dijon"}');
      return compile(input, options, hb);
    };
    hb.precompile = function (input, options) {instrumentation_log('{"type":"FunctionExpression","bodyRange":[89244,89296],"range":[89218,89296],"file":"todomvc/examples.lacunized.instrumented/dijon/node_modules/handlebars/dist/handlebars.js","index":188,"label":"dijon"}');
      return precompile(input, options, hb);
    };

    hb.AST = AST;
    hb.Compiler = Compiler;
    hb.JavaScriptCompiler = JavaScriptCompiler;
    hb.Parser = Parser;
    hb.parse = parse;

    return hb;
  };

  Handlebars = create();
  Handlebars.create = create;

  __exports__ = Handlebars;
  return __exports__;
})(__module1__, __module7__, __module8__, __module10__, __module11__);

  return __module0__;
})();
